// sh-form-control/esm2015/lib/utilities/MaxByteLengthHistory.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { MaxByteLengthService } from "../services/max-bytelength.service";
export class MaxByteLengthHistory {
    /**
     * @param {?} value
     * @param {?} startPoint
     * @param {?} maxByteLength
     */
    constructor(value, startPoint, maxByteLength) {
        this.enterableBytes = 0;
        this.service = new MaxByteLengthService();
        this.cache = value;
        this.startPoint = value.length == 1 && startPoint == 1 ? 0 : startPoint;
        this.maxByteLength = maxByteLength;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    static isAllowOperation(e) {
        return MaxByteLengthService.isAllowOperation(e);
    }
    /**
     * \@ enterableBytesを書き換えます
     * @param {?} value
     * @return {?}
     */
    changeEnterableBytes(value) {
        /** @type {?} */
        const info = this.service.generateByteInfo(value);
        /** @type {?} */
        const entarableByte = this.maxByteLength - info.length;
        this.enterableBytes = entarableByte >= 0 ? entarableByte : 0;
    }
    /**
     * \@ 残りの入力可能byte数を返す
     * @return {?}
     */
    getEnterableBytes() {
        return this.enterableBytes;
    }
    /**
     * \@ inputにセットする値を返す
     * @param {?} data
     * @param {?} value
     * @return {?}
     */
    getResult(data, value) {
        if (!value)
            return "";
        // // 許容バイト数がまだある時
        if (this.enterableBytes)
            return value;
        if (this.startPoint === 0) {
            const { bytes } = this.service.generateByteInfo(value);
            /** @type {?} */
            const maxByteLength = this.service.calcMaxLength(bytes, this.maxByteLength);
            return value.substr(0, maxByteLength);
        }
        const { bytes } = this.service.generateByteInfo(data);
        const { length } = this.service.generateByteInfo(this.cache);
        /** @type {?} */
        const len = this.service.calcMaxLength(bytes, this.maxByteLength, length);
        /** @type {?} */
        const insertStr = data.substr(0, len);
        /** @type {?} */
        const h = this.cache.substr(0, this.startPoint);
        /** @type {?} */
        const f = this.cache.substring(this.startPoint);
        return h + insertStr + f;
    }
    /**
     * アルファベットや数字の入力時に標準のmaxlengthに設定するための値を返す
     * @param {?} value
     * @return {?}
     */
    getMaxLength(value) {
        const { bytes } = this.service.generateByteInfo(value);
        return this.service.calcMaxLength(bytes, this.maxByteLength);
    }
    /**
     * デバッグ用
     * @return {?}
     */
    toString() {
        return JSON.stringify({
            cache: this.cache,
            startPoint: this.startPoint,
            enterableBytes: this.enterableBytes,
            maxByteLength: this.maxByteLength,
        });
    }
}
if (false) {
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.cache;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.startPoint;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.enterableBytes;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.maxByteLength;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.service;
}
