// sh-form-control/esm2015/lib/utilities/sh-app.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ShCtlConst } from "../constShCtl";
import { ShHttpClientConst } from 'sh-http-client';
/**
 * キー制御の初期化など
 */
export class ShAppComponent {
    /**
     * コンストラクタ
     * @param {?} redirectService
     */
    constructor(redirectService) {
        this.redirectService = redirectService;
        this.init();
    }
    /**
     * コンポーネント初期化
     * @return {?}
     */
    ngOnInit() { }
    /**
     * 初期化
     * @private
     * @return {?}
     */
    init() {
        this.redirect();
        this.keyValidation();
        this.setToekn();
    }
    /**
     * @private
     * @return {?}
     */
    redirect() {
        if (this.redirectService.redirectPath) {
            this.redirectService.redirect();
        }
    }
    /**
     * @private
     * @return {?}
     */
    setToekn() {
        /** @type {?} */
        const params = new URLSearchParams(location.search);
        /** @type {?} */
        const token = params.get('access_token');
        if (token) {
            sessionStorage.setItem(`${ShHttpClientConst.AUTHORIZATION_TOKEN}`, token);
            params.delete("access_token");
            history.pushState(null, null, location.pathname);
        }
    }
    /**
     * \@ エンター時にフォーカスを当てたくない場合の条件
     * @private
     * @param {?} elm
     * @return {?}
     */
    isTargetElement(elm) {
        if (elm.type == 'hidden')
            return false;
        if (elm.style.display == 'none')
            return false;
        if (elm.style.opacity == '0')
            return false;
        if (elm.disabled)
            return false;
        /** @type {?} */
        const shctl = this.getShCtlElement(elm);
        if (this.checkShCtlAttr(shctl.attributes, [
            'hidden'
        ]))
            return false;
        return true;
    }
    /**
     * @private
     * @param {?} map
     * @param {?} targets
     * @return {?}
     */
    checkShCtlAttr(map, targets) {
        return targets.map((/**
         * @param {?} t
         * @return {?}
         */
        (t) => {
            return map.getNamedItem(t) != null;
        })).includes(true);
    }
    /**
     * @private
     * @param {?} elm
     * @return {?}
     */
    getShCtlElement(elm) {
        /** @type {?} */
        let parent = elm.parentElement;
        if (!parent)
            return null;
        if (parent.tagName.includes('SH-CTL')) {
            return parent;
        }
        else {
            return this.getShCtlElement(parent);
        }
    }
    /**
     * キーバリデーション付与
     * @return {?}
     */
    keyValidation() {
        window.document.onkeydown = (/**
         * @param {?} event
         * @return {?}
         */
        (event) => {
            if (event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F1 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F2 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F3 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F4 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F5 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F6 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F10 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F11 ||
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_F12) {
                return false;
            }
            if (event.altKey &&
                (event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_LEFT ||
                    event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_RIGHT)) {
                return false;
            }
            if ((event.ctrlKey || event.metaKey) &&
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_N) {
                return false;
            }
            if ((event.ctrlKey || event.metaKey) &&
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_R) {
                return false;
            }
            if ((event.ctrlKey || event.metaKey) &&
                event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_D) {
                return false;
            }
            if (event.keyCode === ShCtlConst.ORDINAL.KEYS.KEY_ENTER) {
                /** @type {?} */
                let input = (/** @type {?} */ (document.activeElement));
                if (input != null) {
                    if (input.type != 'textarea') {
                        //次の操作可能なタグに切り替える
                        /** @type {?} */
                        const nodeList = window.document.querySelectorAll('.can-moved-with-the-enter-key');
                        /** @type {?} */
                        const filteredInput = Array.from(nodeList).filter((/**
                         * @param {?} el
                         * @return {?}
                         */
                        (el) => {
                            return this.isTargetElement((/** @type {?} */ (el)));
                        }));
                        /** @type {?} */
                        let preInput = null;
                        for (let i = 0; i < filteredInput.length; i++) {
                            /** @type {?} */
                            let cur_input = (/** @type {?} */ (filteredInput[i]));
                            if (input.name === cur_input.name) {
                                if (++i >= filteredInput.length) {
                                    i = 0;
                                }
                                cur_input.blur();
                                if (event.shiftKey) {
                                    preInput.focus();
                                }
                                else {
                                    /** @type {?} */
                                    let nextInput = (/** @type {?} */ (filteredInput[i]));
                                    nextInput.focus();
                                }
                                return false;
                            }
                            preInput = (/** @type {?} */ (filteredInput[i]));
                        }
                    }
                }
            }
        });
    }
}
if (false) {
    /**
     * @type {?}
     * @private
     */
    ShAppComponent.prototype.redirectService;
}
