// sh-form-control/esm2015/lib/utilities/ShFormControlType.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class ShFormControlType {
    //コンストラクタでctlNameを必ず指定する
    /**
     * @param {?} _ctlName
     */
    constructor(_ctlName) {
        //要素コントロール名
        this.ctlName = "SHFormControl";
        //表示レイアウト切り替えフラグ(true:閲覧専用/false:編集可能)
        this.isDisabled = false;
        //値バインド変数
        this.value = "";
        //タイトル名
        this.title = "";
        //コントロール挙動(auto/always/nevwer)
        this.floatLabel = "auto";
        //閲覧専用画面でのラベル幅(px)
        this.labelWidth = "150";
        //input要素に対するstyleの指定(形式：{ color: "red" })
        this.styles = {};
        //input要素に対するclassの指定(形式：{ "className": true } or "className className2")
        this.classes = {};
        //input要素に対する必須入力フラグ
        this.required = false;
        //input要素に対する状態の指定
        this.state = "";
        //input要素に対する文言の指定
        this.message = [""];
        this.ctlName = _ctlName;
    }
}
if (false) {
    /** @type {?} */
    ShFormControlType.prototype.ctlName;
    /** @type {?} */
    ShFormControlType.prototype.isDisabled;
    /** @type {?} */
    ShFormControlType.prototype.value;
    /** @type {?} */
    ShFormControlType.prototype.title;
    /** @type {?} */
    ShFormControlType.prototype.floatLabel;
    /** @type {?} */
    ShFormControlType.prototype.labelWidth;
    /** @type {?} */
    ShFormControlType.prototype.maxlength;
    /** @type {?} */
    ShFormControlType.prototype.styles;
    /** @type {?} */
    ShFormControlType.prototype.classes;
    /** @type {?} */
    ShFormControlType.prototype.required;
    /** @type {?} */
    ShFormControlType.prototype.state;
    /** @type {?} */
    ShFormControlType.prototype.message;
}
