// sh-http-client/esm5/lib/utilities/ShMessage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
var SHMessageUtl = /** @class */ (function () {
    function SHMessageUtl() {
    }
    /**
    * テンプレート文字列　形式：{0}{1}{2}
    */
    /**
     * テンプレート文字列　形式：{0}{1}{2}
     * @param {?} text
     * @param {...?} values
     * @return {?}
     */
    SHMessageUtl.prototype.stringFormat = /**
     * テンプレート文字列　形式：{0}{1}{2}
     * @param {?} text
     * @param {...?} values
     * @return {?}
     */
    function (text) {
        var values = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            values[_i - 1] = arguments[_i];
        }
        /** @type {?} */
        var result = text;
        values.forEach((/**
         * @param {?} value
         * @param {?} i
         * @return {?}
         */
        function (value, i) {
            result = result.split("{" + i + "}").join(value);
        }));
        return result;
    };
    return SHMessageUtl;
}());
export { SHMessageUtl };
var SHMessageConfirm = /** @class */ (function (_super) {
    tslib_1.__extends(SHMessageConfirm, _super);
    function SHMessageConfirm() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
    * 実行確認メッセージ
    */
    /**
     * 実行確認メッセージ
     * @param {?} text
     * @return {?}
     */
    SHMessageConfirm.prototype.execution = /**
     * 実行確認メッセージ
     * @param {?} text
     * @return {?}
     */
    function (text) {
        return this.stringFormat("{0}を行います。\nよろしいですか？", text);
    };
    /**
    * 終了確認メッセージ
    */
    /**
     * 終了確認メッセージ
     * @param {?} text
     * @return {?}
     */
    SHMessageConfirm.prototype.finalization = /**
     * 終了確認メッセージ
     * @param {?} text
     * @return {?}
     */
    function (text) {
        return this.stringFormat("{0}を終了します。\nよろしいですか？", text);
    };
    /**
    * 登録確認メッセージ
    */
    /**
     * 登録確認メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.creatingRecord = /**
     * 登録確認メッセージ
     * @return {?}
     */
    function () {
        return this.execution("登録");
    };
    /**
    * 修正確認メッセージ
    */
    /**
     * 修正確認メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.editingRecord = /**
     * 修正確認メッセージ
     * @return {?}
     */
    function () {
        return this.execution("修正");
    };
    /**
    * 削除確認メッセージ
    */
    /**
     * 削除確認メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.deletingRecord = /**
     * 削除確認メッセージ
     * @return {?}
     */
    function () {
        return this.execution("削除");
    };
    /**
    * 更新確認メッセージ
    */
    /**
     * 更新確認メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.updatingRecord = /**
     * 更新確認メッセージ
     * @return {?}
     */
    function () {
        return this.execution("更新");
    };
    /**
    * 印刷確認メッセージ
    */
    /**
     * 印刷確認メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.printingOut = /**
     * 印刷確認メッセージ
     * @return {?}
     */
    function () {
        return this.execution("印刷");
    };
    /**
    * 編集内容が失われる警告メッセージ
    */
    /**
     * 編集内容が失われる警告メッセージ
     * @return {?}
     */
    SHMessageConfirm.prototype.editingContentWillBeLost = /**
     * 編集内容が失われる警告メッセージ
     * @return {?}
     */
    function () {
        return "編集内容が失われます。\nよろしいですか？";
    };
    return SHMessageConfirm;
}(SHMessageUtl));
export { SHMessageConfirm };
var SHMessageFatal = /** @class */ (function (_super) {
    tslib_1.__extends(SHMessageFatal, _super);
    function SHMessageFatal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
    * システムエラーメッセージ
    */
    /**
     * システムエラーメッセージ
     * @return {?}
     */
    SHMessageFatal.prototype.systemErrorHasOccurred = /**
     * システムエラーメッセージ
     * @return {?}
     */
    function () {
        return "システムエラーが発生しました。システム管理者に連絡してください。";
    };
    return SHMessageFatal;
}(SHMessageUtl));
export { SHMessageFatal };
var SHMessageInfo = /** @class */ (function (_super) {
    tslib_1.__extends(SHMessageInfo, _super);
    function SHMessageInfo() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
    * 前のページに戻る場合の確認メッセージ
    */
    /**
     * 前のページに戻る場合の確認メッセージ
     * @param {?} btnName
     * @return {?}
     */
    SHMessageInfo.prototype.dontUseHistoryBack = /**
     * 前のページに戻る場合の確認メッセージ
     * @param {?} btnName
     * @return {?}
     */
    function (btnName) {
        return this.stringFormat("前のページに戻る場合、{0}ボタンから戻ってください。", btnName);
    };
    return SHMessageInfo;
}(SHMessageUtl));
export { SHMessageInfo };
var SHMessage = /** @class */ (function () {
    function SHMessage() {
        /**
         * Confirm
         */
        this.confirm = new SHMessageConfirm();
        /**
         * Fatal
         */
        this.fatal = new SHMessageFatal();
        /**
         * Info
         */
        this.info = new SHMessageInfo();
    }
    return SHMessage;
}());
export { SHMessage };
if (false) {
    /**
     * Confirm
     * @type {?}
     */
    SHMessage.prototype.confirm;
    /**
     * Fatal
     * @type {?}
     */
    SHMessage.prototype.fatal;
    /**
     * Info
     * @type {?}
     */
    SHMessage.prototype.info;
}
