// sh-form-control/esm2015/lib/components/sh-ctl-tan-code/sh-ctl-tan-code.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, } from "@angular/core";
import { SHCtlCommon } from "../sh-ctl-common";
import { MatDialog } from "@angular/material";
import { SHFormPopup } from "../../utilities/ShFormPopup";
import { SmplPopupService } from "../../services/smpl-popup.service";
import { SHCtlPopupDialogComponent } from "../sh-ctl-popup-dialog/sh-ctl-popup-dialog.component";
import { ShCtlConst } from "../../constShCtl";
import { BindCtlService } from "../../services/bind-ctl.service";
import { LockScreenService } from "../../services/lock-screen.service";
export class SHCtlTanCodeComponent extends SHFormPopup {
    /**
     * @param {?} smplPopupService
     * @param {?} dialog
     * @param {?} bindCtlService
     * @param {?} lockScreenService
     */
    constructor(smplPopupService, dialog, bindCtlService, lockScreenService) {
        super(smplPopupService, dialog, bindCtlService, lockScreenService);
        this.smplPopupService = smplPopupService;
        this.dialog = dialog;
        this.bindCtlService = bindCtlService;
        this.lockScreenService = lockScreenService;
        this.obj = ".SHCtlTanCode";
        this.enabledHalfKanaInput = true;
        this.ctlName = "SHCtlTanCode";
        this.maxlength = 6;
    }
    /**
     * コンポーネントに値がバインドされた後で呼び出される
     * プロパティの初期化
     * @return {?}
     */
    ngOnInit() {
        this.onInit();
        if (this.popupWidth) {
            this.extendWidth = this.popupWidth;
        }
    }
    /**
     * onInput
     * @param {?} event input
     * @return {?}
     */
    onInput(event) {
        event.target.value = event.target.value.replace(/[^0-9]/g, "");
        SHCtlCommon.sliceLength(event, this.maxlength);
        if (!this.enabledHalfKanaInput) {
            SHCtlCommon.replaceFullWidth(event);
        }
        event.target.value = event.target.value.replace(/["！”＃＄％＆’（）＝～｜‘｛＋＊｝＜＞？＿－＾￥＠「；：」、。・"]/g, "");
        if (event.target.value.length >= 6 &&
            event.inputType != "deleteContentBackward") {
            if (isNaN(parseInt(event.target.value)) &&
                event.target.value.length != 7) {
                return;
            }
            return;
        }
        if (event.inputType !== "insertCompositionText") {
            SHCtlCommon.replaceSpace(event);
            SHCtlCommon.replaceSign(event);
        }
    }
    /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    onChange(event) {
        this.onBeforeChange(event);
        /** @type {?} */
        let flg = false;
        if (this.noChangeFlg == `OFF`) {
            flg = true;
            if (this.SmplPopupService.getMasterData(this.apiUrl).length == 0 &&
                event.target.value) {
                this.fetchCodes(this.apiUrl).then((/**
                 * @param {?} value
                 * @return {?}
                 */
                (value) => {
                    this.valueChange.emit(value);
                    event.target.value = this.toText(event.target.value);
                    setTimeout((/**
                     * @return {?}
                     */
                    () => {
                        this.onAfterChange(event);
                    }), 0);
                }));
            }
            else {
                this.codes = Object(this.SmplPopupService.getMasterData(this.apiUrl)).data;
                event.target.value = this.setName(event.target.value);
                this.valueChange.emit(event.target.value);
                setTimeout((/**
                 * @return {?}
                 */
                () => {
                    this.onAfterChange(event);
                }), 0);
            }
        }
        if (!flg) {
            setTimeout((/**
             * @return {?}
             */
            () => {
                // this.noChangeFlg == `OFF`;
                setTimeout((/**
                 * @return {?}
                 */
                () => {
                    this.noChangeFlg = `OFF`;
                }), 1000);
                this.valueChange.emit(event.target.value);
                this.onAfterChange(event);
                event.target.value = this.toText(event.target.value);
            }), 0);
        }
    }
    /**
     * ダイアログを開いて閉じる時に値を親に返す
     * @return {?}
     */
    openDialog() {
        /** @type {?} */
        let extendWidth;
        if (this.codes) {
            if (!this.popupWidth &&
                this.codes.length != 0 &&
                Object.values(this.codes[0]).length >= 4) {
                this.extendWidth = ShCtlConst.DIALOG_MAX_WIDTH;
            }
            /** @type {?} */
            let rawCodes;
            /** @type {?} */
            let rawCodesArray = [];
            for (let i = 0; this.codes.length > i; i++) {
                rawCodes = this.codes[i].contactCode;
                rawCodesArray.push(rawCodes);
                /** @type {?} */
                let code = Object.values(this.codes[i])[this.bindColumnKey - 1].toString()
                    .substr(2);
                this.codes[i].contactCode = code;
            }
            this.dialogRef = this.dialog.open(SHCtlPopupDialogComponent, {
                height: ShCtlConst.DIALOG_HEIGHT,
                width: extendWidth,
                data: {
                    dataTitle: this.dataTitle,
                    codes: this.codes,
                    codesHeader: this.codesHeader,
                    hiddenColumns: this.hiddenColumns,
                },
            });
            this.dialogRef.afterClosed().subscribe((/**
             * @param {?} result
             * @return {?}
             */
            (result) => {
                if (result) {
                    this.setValueName(result);
                    this.bindCtlService.setNoChangeFlg({
                        bindColumnArray: this.bindColumnArray,
                        record: result,
                    });
                    this.bindCtlService.setBindValue({
                        bindColumnArray: this.bindColumnArray,
                        record: result,
                    });
                    this.noChangeFlg = `ON`;
                    this.valueChange.emit(this.extractBindValue(result));
                    this.ownValueInputElm.dispatchEvent(new Event("change"));
                }
            }));
        }
    }
    /**
     * 選択された区分コードに紐づく区分名を設定
     * @param {?} val
     * @return {?}
     */
    setName(val) {
        /** @type {?} */
        let record;
        val = this.toText(val);
        if (val.length == 6) {
            record = this.codes.find((/**
             * @param {?} element
             * @return {?}
             */
            (element) => this.toText(Object.values(element)[this.bindColumnKey - 1]).includes(val)));
        }
        if (record != null) {
            /** @type {?} */
            const name = (/** @type {?} */ (Object.values(record)[this.bindColumnValue - 1]));
            if (name && val.length == 6) {
                this.setValueName(record);
                this.noChangeFlg = "ON";
                this.bindCtlService.setBindValue({
                    bindColumnArray: this.bindColumnArray,
                    record,
                });
                this.bindCtlService.setNoChangeFlg({
                    bindColumnArray: this.bindColumnArray,
                    record,
                });
                val = Object.values(record)[this.bindColumnKey - 1];
                /** @type {?} */
                const inputElement = this.ownValueInputElm;
                if (inputElement) {
                    setTimeout((/**
                     * @return {?}
                     */
                    () => {
                        inputElement.value = this.toText(val);
                    }), 0);
                }
            }
        }
        else {
            val = ``;
        }
        return val;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    checkValue(value) {
        if (!value)
            return false;
        if (value.length < 6) {
            this.clearValue();
            return false;
        }
        //let matchedArray = [];
        if (this.codes[0]) {
            /** @type {?} */
            const filterResult = this.codes.filter((/**
             * @param {?} element
             * @return {?}
             */
            (element) => 
            //this.hasMatchRecords(element, value)
            ((/** @type {?} */ (Object.values(element)[this.bindColumnKey - 1]))).includes(value)));
            if (filterResult.length == 0) {
                this.clearValue();
                return false;
            }
            if (filterResult.length >= 2) {
                alert(value + "に複数の候補が検出されました");
                this.clearValue();
                this.valueChange.emit("");
                return false;
            }
        }
        return true;
    }
    /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    onClick(event) {
        this.onBeforeClick(event);
        this.onAfterClick(event);
    }
    /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    onFocus(event) {
        this.onBeforeFocus(event);
        event.target.value = this.toText(event.target.value);
        event.target.select();
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.onAfterFocus(event);
        }), 0);
    }
    /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    onBlur(event) {
        this.onBeforeBlur(event);
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.onAfterBlur(event);
        }), 0);
    }
    /**
     * onKeyUp
     * @param {?} event KeyUp
     * @return {?}
     */
    onKeyUp(event) {
        if (!event.isComposing) {
            event.target.value = this.toHankaku(this.hiraganaToKatagana(event.target.value));
        }
    }
    /**
     * 先頭除去
     * @param {?} inputTargetValue event.target.value
     * @return {?}
     */
    toText(inputTargetValue) {
        if (SHCtlCommon.isBlank(inputTargetValue)) {
            return "";
        }
        if (inputTargetValue.length === 8) {
            return inputTargetValue.substr(2);
        }
        return inputTargetValue;
    }
    /**
     * ひらがなをカタカナに変換
     * @param {?} inputValue
     * @return {?}
     */
    hiraganaToKatagana(inputValue) {
        return inputValue.replace(/[\u3041-\u3096]/g, (/**
         * @param {?} match
         * @return {?}
         */
        function (match) {
            /** @type {?} */
            var chr = match.charCodeAt(0) + 0x60;
            return String.fromCharCode(chr);
        }));
    }
    /**
     * 入力文字を半角カナに変換
     * @param {?} inputValue
     * @return {?}
     */
    toHankaku(inputValue) {
        /** @type {?} */
        let half = "0123456789";
        half += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        half += "abcdefghijklmnopqrstuvwxyz";
        half += "｡､｢\\｣;:･-+|_@., ";
        half += "ｱｲｳｴｵｶｷｸｹｺｻｼｽｾｿﾀﾁﾂﾃﾄﾅﾆﾇﾈﾉﾊﾋﾌﾍﾎﾏﾐﾑﾒﾓﾔﾕﾖﾗﾘﾙﾚﾛﾜｦﾝｧｨｩｪｫｯｬｭｮｰ";
        /** @type {?} */
        let halfArr = new Array("ｳﾞ", "ｶﾞ", "ｷﾞ", "ｸﾞ", "ｹﾞ", "ｺﾞ", "ｻﾞ", "ｼﾞ", "ｽﾞ", "ｾﾞ", "ｿﾞ", "ﾀﾞ", "ﾁﾞ", "ﾂﾞ", "ﾃﾞ", "ﾄﾞ", "ﾊﾞ", "ﾋﾞ", "ﾌﾞ", "ﾍﾞ", "ﾎﾞ", "ﾊﾟ", "ﾋﾟ", "ﾌﾟ", "ﾍﾟ", "ﾎﾟ");
        /** @type {?} */
        let full = "０１２３４５６７８９";
        full += "ＡＢＣＤＥＦＧＨＩＪＫＬＭＮＯＰＱＲＳＴＵＶＷＸＹＺ";
        full += "ａｂｃｄｅｆｇｈｉｊｋｌｍｎｏｐｑｒｓｔｕｖｗｘｙｚ";
        full += "。、「￥」；：・－＋｜＿＠．，　";
        full +=
            "アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホマミムメモヤユヨラリルレロワヲンァィゥェォッャュョー";
        /** @type {?} */
        let fullArr = new Array("ヴ", "ガ", "ギ", "グ", "ゲ", "ゴ", "ザ", "ジ", "ズ", "ゼ", "ゾ", "ダ", "ヂ", "ヅ", "デ", "ド", "バ", "ビ", "ブ", "ベ", "ボ", "パ", "ピ", "プ", "ペ", "ポ");
        /** @type {?} */
        let str = "";
        //半角カナ用
        for (let i = 0; i < halfArr.length; i++) {
            /** @type {?} */
            let reg = new RegExp(fullArr[i], "g");
            inputValue = inputValue.replace(reg, halfArr[i]);
        }
        for (let i = 0; i < inputValue.length; i++) {
            /** @type {?} */
            let oneStr = inputValue.charAt(i);
            /** @type {?} */
            let num = full.indexOf(oneStr, 0);
            oneStr = num >= 0 ? half.charAt(num) : oneStr;
            str += oneStr;
        }
        return str;
    }
}
SHCtlTanCodeComponent.decorators = [
    { type: Component, args: [{
                selector: "sh-ctl-tan-code",
                template: "<ng-container *ngIf=\"!onlyField; else onlyFieldControl\">\r\n  <div *ngIf=\"!isDisabled\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <input class=\"bind-flg__input\" [(value)]=\"noChangeFlg\" [name]=\"noChangeFlgCtlName\" (change)=\"noChangeFlgChange($event)\"/>\r\n    <div class=\"sh-flex sh-flex-center\">\r\n      <mat-form-field style=\"width: 33.3%;\" class=\"mat-validation\" [floatLabel]=\"floatLabel\">\r\n        <input #inputElm matInput [required]=\"required\" (input)=\"onInput($event)\" (click)=\"onClick($event)\"\r\n          [(ngModel)]=\"value\" class=\"SHCtlJyuCode can-moved-with-the-enter-key  {{strClass}}\" [disabled]=\"disabled\"\r\n          (change)=\"onChange($event)\" (focus)=\"onFocus($event)\" maxlength=\"8\" type=\"tel\"\r\n          [placeholder]=\"title\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" attr.name=\"{{ctlName}}\" />\r\n        <button mat-button *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n          aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n          <mat-icon>close</mat-icon>\r\n        </button>\r\n      </mat-form-field>\r\n      <mat-form-field style=\"width: 66.6%;\" class=\"result-container\">\r\n        <input class=\"result-input\" matInput type=\"tel\" [(ngModel)]=\"name\" [ngStyle]=\"styles\" [ngClass]=\"classes\"\r\n          attr.name=\"{{popupLabelName}}\" disabled>\r\n        <mat-icon mat-button matSuffix class=\"mat-search-btn\" *ngIf=\"!inputElm.readOnly && !inputElm.disabled\"\r\n          (click)=\"getCodes()\">search</mat-icon>\r\n      </mat-form-field>\r\n    </div>\r\n    <span class=\"custom-alert-message\">{{message.toString().split(',').join(' ')}}</span>\r\n  </div>\r\n</ng-container>\r\n\r\n<ng-template #onlyFieldControl>\r\n  <div class=\"sh-flex sh-flex-center\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <div class=\"only-field\" style=\"width: 33.3%;\">\r\n      <input class=\"bind-flg__input\" [(value)]=\"noChangeFlg\" [name]=\"noChangeFlgCtlName\" (change)=\"noChangeFlgChange($event)\"/>\r\n      <input #inputElm matInput [required]=\"required\" (input)=\"onInput($event)\" (click)=\"onClick($event)\"\r\n        [(ngModel)]=\"value\" class=\"SHCtlJyuCode can-moved-with-the-enter-key  {{strClass}}\" [disabled]=\"disabled\"\r\n        (change)=\"onChange($event)\" (focus)=\"onFocus($event)\" maxlength=\"8\" type=\"tel\"\r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\" attr.name=\"{{ctlName}}\" />\r\n      <button mat-button *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n        aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n        <mat-icon>close</mat-icon>\r\n      </button>\r\n    </div>\r\n    <div class=\"only-field only-label\" style=\"width: 66.6%;\">\r\n      <input class=\"result-input\" matInput type=\"tel\" [(ngModel)]=\"name\" [ngStyle]=\"styles\" [ngClass]=\"classes\"\r\n        attr.name=\"{{popupLabelName}}\" disabled>\r\n      <mat-icon mat-button matSuffix class=\"mat-search-btn\" *ngIf=\"!inputElm.readOnly && !inputElm.disabled\"\r\n        (click)=\"getCodes()\">search</mat-icon>\r\n      </div>\r\n    </div>\r\n  </ng-template>\r\n\r\n<div *ngIf=\"isDisabled\" class=\"display-ctl-view\">\r\n  <div>\r\n    <div *ngIf=\"title\" class=\"display-ctl-view\">\r\n      <div class=\"display-result\" [ngStyle]=\"{'width.px': labelWidth}\" fxLayout=\"row wrap\" fxFlex>\r\n        <div fxFlex class=\"item-title\"><strong>{{title}}:</strong></div>\r\n      </div>\r\n      <div class=\"valueOutput-container\">\r\n        <div fxFlex class=\"valueOutput\">{{toText(value)}}</div>\r\n        <div fxFlex class=\"valueOutput-second\">{{name}}</div>\r\n      </div>\r\n    </div>\r\n\r\n    <div *ngIf=\"!title\" class=\"display-ctl-view notitle-value\">\r\n      <div class=\"valueOutput-container\">\r\n        <div fxFlex class=\"valueOutput\">{{value}}</div>\r\n        <div fxFlex class=\"valueOutput-second\">{{name}}</div>\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                styles: [".input-group-text{background-color:#fff;border:none}.btn{border:none}.btn.btn-primary:disabled{background-color:#999}.btn-primary{background-color:#17a2b8}.btn-primary:hover{background-color:#1792b8}.fixed-bottom{top:708px;height:65px;max-width:1024px;margin-right:auto;margin-left:auto}.mat-search-btn{color:#bababa;font-size:1.1rem}.mat-search-btn:hover{cursor:pointer}.result-input{text-overflow:ellipsis}.display-ctl-view{display:-webkit-box;display:flex}.display-result{margin-left:30px}.valueOutput-container{display:-webkit-box;display:flex;-webkit-box-flex:1;flex:1}.valueOutput{margin-right:14px;padding-top:4px}.notitle-value{width:100%;margin-left:30px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
            }] }
];
/** @nocollapse */
SHCtlTanCodeComponent.ctorParameters = () => [
    { type: SmplPopupService },
    { type: MatDialog },
    { type: BindCtlService },
    { type: LockScreenService }
];
SHCtlTanCodeComponent.propDecorators = {
    enabledHalfKanaInput: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    SHCtlTanCodeComponent.prototype.obj;
    /** @type {?} */
    SHCtlTanCodeComponent.prototype.enabledHalfKanaInput;
    /**
     * @type {?}
     * @protected
     */
    SHCtlTanCodeComponent.prototype.smplPopupService;
    /**
     * @type {?}
     * @protected
     */
    SHCtlTanCodeComponent.prototype.dialog;
    /**
     * @type {?}
     * @protected
     */
    SHCtlTanCodeComponent.prototype.bindCtlService;
    /**
     * @type {?}
     * @protected
     */
    SHCtlTanCodeComponent.prototype.lockScreenService;
}
