// sh-form-control/esm5/lib/sh-ctl-common.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
//Angular Material
import { MatFormFieldModule, MatInputModule, MatDialogModule, MatTableModule, MatButtonModule, MatButtonToggleModule, MatRadioModule, MatSelectModule, MatDatepickerModule, MatNativeDateModule, MatIconModule, MatCheckboxModule, MatProgressSpinnerModule, } from '@angular/material';
import { SHCtlAllCharComponent } from './components/sh-ctl-all-char/sh-ctl-all-char.component';
import { SHCtlAllCharMLComponent } from './components/sh-ctl-all-char-ml/sh-ctl-all-char-ml.component';
import { SHCtlBunjNoComponent } from './components/sh-ctl-bunj-no/sh-ctl-bunj-no.component';
import { SHCtlDateComponent } from './components/sh-ctl-date/sh-ctl-date.component';
import { SHCtlHalfCharComponent } from './components/sh-ctl-half-char/sh-ctl-half-char.component';
import { SHCtlJigCodeComponent } from './components/sh-ctl-jig-code/sh-ctl-jig-code.component';
import { SHCtlPopupDialogComponent } from './components/sh-ctl-popup-dialog/sh-ctl-popup-dialog.component';
import { SHCtlPopupDialogMultiComponent } from './components/sh-ctl-popup-multi-dialog/sh-ctl-popup-multi-dialog.component';
import { SHCtlJyuCodeComponent } from './components/sh-ctl-jyu-code/sh-ctl-jyu-code.component';
import { SHCtlKatakanaComponent } from './components/sh-ctl-katakana/sh-ctl-katakana.component';
import { SHCtlNumberComponent } from './components/sh-ctl-number/sh-ctl-number.component';
import { SHCtlSesyuCodeComponent } from './components/sh-ctl-sesyu-code/sh-ctl-sesyu-code.component';
import { SHCtlTanCodeComponent } from './components/sh-ctl-tan-code/sh-ctl-tan-code.component';
import { SHCtlTelNoComponent } from './components/sh-ctl-tel-no/sh-ctl-tel-no.component';
import { SHCtlTimeComponent } from './components/sh-ctl-time/sh-ctl-time.component';
import { SHCtlTokuCodeComponent } from './components/sh-ctl-toku-code/sh-ctl-toku-code.component';
import { SHCtlWideCharComponent } from './components/sh-ctl-wide-char/sh-ctl-wide-char.component';
import { SHCtlWideCharMLComponent } from './components/sh-ctl-wide-char-ml/sh-ctl-wide-char-ml.component';
import { SHCtlYearMonthComponent } from './components/sh-ctl-year-month/sh-ctl-year-month.component';
import { SHCtlZipNoComponent } from './components/sh-ctl-zip-no/sh-ctl-zip-no.component';
import { SHCtlKbnRadioComponent } from './components/sh-ctl-kbn-radio/sh-ctl-kbn-radio.component';
import { SHCtlKbnSelectComponent } from './components/sh-ctl-kbn-select/sh-ctl-kbn-select.component';
import { ShCtlPopupComponent } from './components/sh-ctl-popup/sh-ctl-popup.component';
import { ShCtlPopupMultiComponent } from './components/sh-ctl-popup-multi/sh-ctl-popup-multi.component';
import { ShCtlValidatedErrorComponent } from './components/sh-ctl-validated-error/sh-ctl-validated-error.component';
import { SHCtlHalfNumberComponent } from './components/sh-ctl-half-number/sh-ctl-half-number.component';
import { SHCtlCheckComponent } from './components/sh-ctl-check/sh-ctl-check.component';
import { SHCtlLockScreenComponent } from './components/sh-ctl-lock-screen/sh-ctl-lock-screen.component';
import { NgDateValuePipe } from './pipes/ng-date-value.pipe';
import { ShHttpClient } from 'sh-http-client';
import { ShFlexRowDirective } from './directives/sh-flex-row.directive';
import { ShFlexDirective } from './directives/sh-flex.directive';
import { ShReadOnlyDirective } from './directives/sh-readonly.directive';
import { ShCtlStateDirective } from './directives/sh-ctl-state.directive';
var ShCtlCommonModule = /** @class */ (function () {
    function ShCtlCommonModule() {
    }
    ShCtlCommonModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        SHCtlAllCharComponent,
                        SHCtlAllCharMLComponent,
                        SHCtlBunjNoComponent,
                        SHCtlDateComponent,
                        SHCtlHalfCharComponent,
                        SHCtlJigCodeComponent,
                        SHCtlJyuCodeComponent,
                        SHCtlKatakanaComponent,
                        SHCtlNumberComponent,
                        SHCtlSesyuCodeComponent,
                        SHCtlTanCodeComponent,
                        SHCtlTelNoComponent,
                        SHCtlTimeComponent,
                        SHCtlTokuCodeComponent,
                        SHCtlWideCharComponent,
                        SHCtlWideCharMLComponent,
                        SHCtlYearMonthComponent,
                        SHCtlZipNoComponent,
                        SHCtlKbnRadioComponent,
                        SHCtlKbnSelectComponent,
                        ShCtlPopupComponent,
                        ShCtlPopupMultiComponent,
                        SHCtlPopupDialogComponent,
                        SHCtlPopupDialogMultiComponent,
                        ShCtlValidatedErrorComponent,
                        SHCtlHalfNumberComponent,
                        SHCtlCheckComponent,
                        SHCtlLockScreenComponent,
                        NgDateValuePipe,
                        ShFlexRowDirective,
                        ShFlexDirective,
                        ShReadOnlyDirective,
                        ShCtlStateDirective,
                    ],
                    entryComponents: [
                        SHCtlPopupDialogComponent,
                        SHCtlPopupDialogMultiComponent,
                    ],
                    imports: [
                        FormsModule,
                        CommonModule,
                        MatInputModule,
                        MatFormFieldModule,
                        ReactiveFormsModule,
                        MatDialogModule,
                        MatTableModule,
                        MatButtonModule,
                        MatButtonToggleModule,
                        MatRadioModule,
                        MatSelectModule,
                        MatDatepickerModule,
                        MatNativeDateModule,
                        MatIconModule,
                        MatCheckboxModule,
                        MatProgressSpinnerModule,
                        ShHttpClient
                    ],
                    exports: [
                        SHCtlAllCharComponent,
                        SHCtlAllCharMLComponent,
                        SHCtlBunjNoComponent,
                        SHCtlDateComponent,
                        SHCtlHalfCharComponent,
                        SHCtlJigCodeComponent,
                        SHCtlJyuCodeComponent,
                        SHCtlKatakanaComponent,
                        SHCtlNumberComponent,
                        SHCtlSesyuCodeComponent,
                        SHCtlTanCodeComponent,
                        SHCtlTelNoComponent,
                        SHCtlTimeComponent,
                        SHCtlTokuCodeComponent,
                        SHCtlWideCharComponent,
                        SHCtlWideCharMLComponent,
                        SHCtlYearMonthComponent,
                        SHCtlZipNoComponent,
                        SHCtlKbnRadioComponent,
                        SHCtlKbnSelectComponent,
                        ShCtlPopupComponent,
                        ShCtlPopupMultiComponent,
                        ShCtlValidatedErrorComponent,
                        SHCtlHalfNumberComponent,
                        SHCtlCheckComponent,
                        SHCtlLockScreenComponent,
                        NgDateValuePipe,
                        ShFlexRowDirective,
                        ShFlexDirective,
                        ShReadOnlyDirective,
                        ShCtlStateDirective
                    ],
                    providers: []
                },] }
    ];
    return ShCtlCommonModule;
}());
export { ShCtlCommonModule };
