// sh-form-control/esm2015/lib/services/lock-screen.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class LockScreenService {
    /**
     * コンストラクタ
     */
    constructor() {
        this.isConnecting = false;
        this.isConnectingSubject = new Subject();
        this.isConnectingState = this.isConnectingSubject.asObservable();
        this.count = 0;
    }
    /**
     * 画面をロックする(ユーザーが操作出来ないようにする)
     * @return {?}
     */
    lockScreen() {
        this.count += 1;
        if (this.count === 1) {
            this.isConnecting = true;
            this.isConnectingSubject.next(this.isConnecting);
        }
    }
    /**
     * 画面のロックを解除する(ユーザーが操作出来る状態にする)
     * @return {?}
     */
    unlock() {
        this.count -= 1;
        if (this.count === 0) {
            this.isConnecting = false;
            this.isConnectingSubject.next(this.isConnecting);
        }
    }
}
LockScreenService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
LockScreenService.ctorParameters = () => [];
/** @nocollapse */ LockScreenService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function LockScreenService_Factory() { return new LockScreenService(); }, token: LockScreenService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    LockScreenService.prototype.isConnecting;
    /**
     * @type {?}
     * @private
     */
    LockScreenService.prototype.isConnectingSubject;
    /** @type {?} */
    LockScreenService.prototype.isConnectingState;
    /** @type {?} */
    LockScreenService.prototype.count;
}
