// sh-form-control/esm2015/lib/services/max-bytelength.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
export class MaxByteLengthService {
    constructor() { }
    /**
     * @param {?} value
     * @return {?}
     */
    generateByteInfo(value) {
        /** @type {?} */
        var length = 0;
        /** @type {?} */
        let bytes = [];
        if (!value)
            return { length, bytes };
        for (var i = 0; i < value.length; i++) {
            /** @type {?} */
            var c = value.charCodeAt(i);
            if ((c >= 0x0 && c < 0x81) ||
                c === 0xf8f0 ||
                (c >= 0xff61 && c < 0xffa0) ||
                (c >= 0xf8f1 && c < 0xf8f4)) {
                length += 1;
                bytes.push(1);
            }
            else {
                length += 2;
                bytes.push(2);
            }
        }
        return { length, bytes };
    }
    /**
     * @param {?} bytes
     * @param {?} maxBytelength
     * @param {?=} total
     * @return {?}
     */
    calcMaxLength(bytes, maxBytelength, total = 0) {
        /** @type {?} */
        let i;
        for (i = 0; i < bytes.length; i++) {
            total += bytes[i];
            if (total >= maxBytelength) {
                break;
            }
        }
        return (total - maxBytelength) > 0 ? i : i + 1;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    static isAllowOperation(e) {
        if (e.ctrlKey && MaxByteLengthService.isAllowWithCtlKey(e.key))
            return true;
        if (e.shiftKey && MaxByteLengthService.isAllowWithShiftKey(e.key))
            return true;
        if (MaxByteLengthService.isAllowCodes(e.code))
            return true;
        if (MaxByteLengthService.isAllowKey(e.key))
            return true;
        return false;
    }
    /**
     * @param {?} key
     * @return {?}
     */
    static isAllowKey(key) {
        if (MaxByteLengthService.allowKeys.includes(key))
            return true;
        return false;
    }
    /**
     * @private
     * @param {?} code
     * @return {?}
     */
    static isAllowCodes(code) {
        if (MaxByteLengthService.allowCodes.includes(code))
            return true;
        return false;
    }
    /**
     * @private
     * @param {?} key
     * @return {?}
     */
    static isAllowWithCtlKey(key) {
        if (MaxByteLengthService.allowWithCtlKey.includes(key))
            return true;
        return false;
    }
    /**
     * @private
     * @param {?} key
     * @return {?}
     */
    static isAllowWithShiftKey(key) {
        if (MaxByteLengthService.allowWithShiftKey.includes(key))
            return true;
        return false;
    }
}
MaxByteLengthService.allowKeys = ["ArrowRight", "ArrowLeft", "Backspace", "ArrowDown", "ArrowUp"];
MaxByteLengthService.allowCodes = ["Enter", "Tab"];
MaxByteLengthService.allowWithCtlKey = ["a", "Home", "End"];
MaxByteLengthService.allowWithShiftKey = ["Home", "End"];
MaxByteLengthService.decorators = [
    { type: Injectable, args: [{
                providedIn: `root`,
            },] }
];
/** @nocollapse */
MaxByteLengthService.ctorParameters = () => [];
/** @nocollapse */ MaxByteLengthService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function MaxByteLengthService_Factory() { return new MaxByteLengthService(); }, token: MaxByteLengthService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthService.allowKeys;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthService.allowCodes;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthService.allowWithCtlKey;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthService.allowWithShiftKey;
}
