// sh-form-control/esm2015/lib/services/sh-ctl-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShHttpClientService } from 'sh-http-client';
import * as i0 from "@angular/core";
import * as i1 from "sh-http-client";
/**
 * ShCtl　Api通信クラス
 */
export class ShCtlApiService {
    /**
     * @param {?} apiComService
     */
    constructor(apiComService) {
        this.apiComService = apiComService;
    }
    /**
     * API返却結果に応じて処理分岐を行い、ApplicationDataのみを返却する
     * @private
     * @template T
     * @param {?} apiResponseBody
     * @return {?} API返却結果のApplicationData
     */
    getApplicationData(apiResponseBody) {
        //バリデーション処理
        //アラート処理
        //システムに返す際は処理に必要なデータだけにする
        return apiResponseBody.applicationData;
    }
    /**
     * GET
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    get(url) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let ret = null;
            yield this.apiComService.get(url, false)
                .then((/**
             * @param {?} response
             * @return {?}
             */
            response => {
                ret = this.getApplicationData(response);
            }));
            return ret;
        });
    }
    /**
     * POST
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    post(url, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let ret = null;
            yield this.apiComService.post(url, data)
                .then((/**
             * @param {?} response
             * @return {?}
             */
            response => {
                ret = this.getApplicationData(response);
            }));
            return ret;
        });
    }
    /**
     * PUT
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    put(url, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let ret = null;
            yield this.apiComService.put(url, data)
                .then((/**
             * @param {?} response
             * @return {?}
             */
            response => {
                ret = this.getApplicationData(response);
            }));
            return ret;
        });
    }
    /**
     * DELETE
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    delete(url) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            let ret = null;
            yield this.apiComService.delete(url)
                .then((/**
             * @param {?} response
             * @return {?}
             */
            response => {
                ret = this.getApplicationData(response);
            }));
            return ret;
        });
    }
}
ShCtlApiService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ShCtlApiService.ctorParameters = () => [
    { type: ShHttpClientService }
];
/** @nocollapse */ ShCtlApiService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ShCtlApiService_Factory() { return new ShCtlApiService(i0.ɵɵinject(i1.ShHttpClientService)); }, token: ShCtlApiService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    ShCtlApiService.prototype.apiComService;
}
