// sh-form-control/esm2015/lib/services/smpl-popup.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShCtlApiService } from './sh-ctl-api.service';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./sh-ctl-api.service";
/**
 * 区分マスタの情報を取得
 */
export class SmplPopupService {
    /**
     * @param {?} apiComService
     */
    constructor(apiComService) {
        this.apiComService = apiComService;
        this.popupCdSubject = new Subject();
        this.popupCdState = this.popupCdSubject.asObservable();
        //汎用マスタ
        this.masterCache = [];
    }
    /**
     * 汎用マスタ取得
     * @param {?} url
     * @return {?}
     */
    fetchAll(url) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.apiComService.get(url);
        });
    }
    /**
     * 指定したマスタのキャッシュしているデータを取得
     * @param {?} key マスタの名前
     * @return {?}
     */
    getMasterData(key) {
        /** @type {?} */
        let cache = this.masterCache.filter((/**
         * @param {?} item
         * @return {?}
         */
        item => {
            return item.masterName == key;
        }));
        return cache.length > 0 ? cache[0].masterCache : [];
    }
    /**
     * マスタデータをキャッシュする
     * @param {?} key マスタの名前
     * @param {?} masterData マスタデータ
     * @return {?}
     */
    setMasterData(key, masterData) {
        if (this.masterCache.length > 0) {
            for (let item of this.masterCache) {
                if (item.masterName == key) {
                    item.masterCache = masterData;
                    break;
                }
            }
        }
        this.masterCache = this.masterCache.concat([{ masterName: key, masterCache: masterData }]);
    }
}
SmplPopupService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
SmplPopupService.ctorParameters = () => [
    { type: ShCtlApiService }
];
/** @nocollapse */ SmplPopupService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function SmplPopupService_Factory() { return new SmplPopupService(i0.ɵɵinject(i1.ShCtlApiService)); }, token: SmplPopupService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    SmplPopupService.prototype.popupCdSubject;
    /** @type {?} */
    SmplPopupService.prototype.popupCdState;
    /**
     * @type {?}
     * @private
     */
    SmplPopupService.prototype.masterCache;
    /**
     * @type {?}
     * @private
     */
    SmplPopupService.prototype.apiComService;
}
