// sh-form-control/esm2015/lib/components/sh-ctl-validated-error/sh-ctl-validated-error.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { ValidationErrorService } from 'sh-http-client';
export class ShCtlValidatedErrorComponent {
    /**
     * コンストラクタ
     * @param {?} validationErrorService
     */
    constructor(validationErrorService) {
        this.validationErrorService = validationErrorService;
        this.isError = false;
    }
    /**
     * コンポーネント初期化
     * @return {?}
     */
    ngOnInit() {
        this.subscription = this.validationErrorService.validationErrorState
            .subscribe((/**
         * @param {?} response
         * @return {?}
         */
        response => {
            if (response.show) {
                this.isError = true;
                this.errors = [];
                Object.keys(response.meesages).forEach((/**
                 * @param {?} controlName
                 * @return {?}
                 */
                controlName => {
                    this.errors = [...this.errors, ...response.meesages[controlName]];
                }));
            }
            else {
                this.isError = false;
                this.errors = [];
            }
        }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
}
ShCtlValidatedErrorComponent.decorators = [
    { type: Component, args: [{
                selector: 'sh-ctl-validated-error',
                template: "<div *ngIf=\"isError\" class=\"error-message\">\r\n  <div *ngFor=\"let item of errors\" class=\"errorMessages text-red\" role=\"alert\">\r\n    \u30FB{{item}}\r\n  </div>\r\n</div>\r\n",
                styles: [".error-message{text-align:left;width:100%;height:100%}.text-red{color:red}"]
            }] }
];
/** @nocollapse */
ShCtlValidatedErrorComponent.ctorParameters = () => [
    { type: ValidationErrorService }
];
ShCtlValidatedErrorComponent.propDecorators = {
    message: [{ type: Input }],
    isError: [{ type: Input }],
    errors: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.message;
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.isError;
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.errors;
    /**
     * @type {?}
     * @private
     */
    ShCtlValidatedErrorComponent.prototype.subscription;
    /**
     * @type {?}
     * @private
     */
    ShCtlValidatedErrorComponent.prototype.validationErrorService;
}
