// sh-form-control/esm5/lib/components/sh-ctl-popup-multi/sh-ctl-popup-multi.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from "@angular/core";
import { SmplPopupService } from "../../services/smpl-popup.service";
import { MatDialog } from "@angular/material";
import { SHFormControl } from '../../utilities/ShFormControl';
import { SHCtlPopupDialogMultiComponent } from '../sh-ctl-popup-multi-dialog/sh-ctl-popup-multi-dialog.component';
import { ShCtlConst } from '../../constShCtl';
var ShCtlPopupMultiComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ShCtlPopupMultiComponent, _super);
    function ShCtlPopupMultiComponent(SmplPopupService, dialog) {
        var _this = _super.call(this) || this;
        _this.SmplPopupService = SmplPopupService;
        _this.dialog = dialog;
        //データタイトル名
        _this.dataTitle = "";
        //API URL
        _this.apiUrl = "/api/";
        //データキャッシュON/OFF機能フラグ
        _this.dataCache = false;
        //キーとなる列番号(起動時のチェックを入れるためのキー判定基準になります。)
        _this.uniqueColumns = [1];
        //ポップアップ上非表示となる列番号
        _this.hiddenColumns = [];
        _this.extendWidth = ShCtlConst.DIALOG_WIDTH;
        //Dialogの結果を受け取る
        _this.dialogRef = null;
        _this.codes = [];
        _this.codesHeader = [];
        _this.ctlName = "SHCtlPopupMulti";
        return _this;
    }
    /**
     * ダイアログを開いて閉じる時に値を親に返す
     */
    /**
     * ダイアログを開いて閉じる時に値を親に返す
     * @return {?}
     */
    ShCtlPopupMultiComponent.prototype.openDialog = /**
     * ダイアログを開いて閉じる時に値を親に返す
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.codes) {
            if (!this.popupWidth && this.codes.length != 0 && Object.values(this.codes[0]).length >= 4) {
                this.extendWidth = ShCtlConst.DIALOG_MAX_WIDTH;
            }
            this.dialogRef = this.dialog.open(SHCtlPopupDialogMultiComponent, {
                height: ShCtlConst.DIALOG_HEIGHT,
                width: this.extendWidth,
                data: {
                    dataTitle: this.dataTitle,
                    value: this.value,
                    uniqueColumns: this.uniqueColumns,
                    hiddenColumns: this.hiddenColumns,
                }
            });
            this.dialogRef.afterClosed().subscribe((/**
             * @param {?} result
             * @return {?}
             */
            function (result) {
                if (result != undefined)
                    _this.valueChange.emit(result);
            }));
        }
    };
    /**
     *プロパティの初期化
     *通常サービスにキャッシュしているが、無ければ再度マスタより取得
     */
    /**
     * プロパティの初期化
     * 通常サービスにキャッシュしているが、無ければ再度マスタより取得
     * @return {?}
     */
    ShCtlPopupMultiComponent.prototype.ngOnInit = /**
     * プロパティの初期化
     * 通常サービスにキャッシュしているが、無ければ再度マスタより取得
     * @return {?}
     */
    function () {
        var _this = this;
        this.onInit();
        if (this.popupWidth) {
            this.extendWidth = this.popupWidth;
        }
        // //ユニークな文字列
        if (this.SmplPopupService.getMasterData(this.apiUrl).length >= 0) {
            this.SmplPopupService.fetchAll(this.apiUrl).then((/**
             * @param {?} response
             * @return {?}
             */
            function (response) {
                _this.codes = response.data;
                if (response.header == null) {
                    return;
                }
                _this.codesHeader = Object.values(response.header);
                _this.dataTitle = response.title;
            }));
        }
    };
    /**
     * マスタ情報よりコードを取得
     */
    /**
     * マスタ情報よりコードを取得
     * @return {?}
     */
    ShCtlPopupMultiComponent.prototype.getCodes = /**
     * マスタ情報よりコードを取得
     * @return {?}
     */
    function () {
        var _this = this;
        this.openDialog();
        if (this.SmplPopupService.getMasterData(this.apiUrl).length == 0) {
            this.SmplPopupService.fetchAll(this.apiUrl).then((/**
             * @param {?} response
             * @return {?}
             */
            function (response) {
                _this.codes = response.data;
                _this.codesHeader = Object.values(response.header);
                _this.dataTitle = response.title;
                if (_this.dataCache) {
                    _this.SmplPopupService.setMasterData(_this.apiUrl, response);
                }
                if (_this.dialogRef.componentInstance) {
                    _this.dialogRef.componentInstance.codesHeader = _this.codesHeader;
                    _this.dialogRef.componentInstance.unchangeableCodes = _this.codes;
                    _this.dialogRef.componentInstance.codes = _this.codes;
                    if (!_this.popupWidth && Object.values(_this.codes[0]).length >= 4) {
                        _this.extendWidth = ShCtlConst.DIALOG_MAX_WIDTH;
                    }
                    /** @type {?} */
                    var cols_1 = ['select'];
                    _this.dialogRef.componentInstance.getHashKeys().forEach((/**
                     * @param {?} name
                     * @param {?} idx
                     * @return {?}
                     */
                    function (name, idx) {
                        if (!_this.hiddenColumns.find((/**
                         * @param {?} _
                         * @return {?}
                         */
                        function (_) { return _ == (idx + 1); })))
                            cols_1.push(name);
                    }));
                    _this.dialogRef.componentInstance.displayedColumns = cols_1;
                }
            }));
        }
        else {
            this.codes = Object(this.SmplPopupService.getMasterData(this.apiUrl)).data;
            this.codesHeader = Object.values(Object(this.SmplPopupService.getMasterData(this.apiUrl)).header);
            if (this.dialogRef.componentInstance) {
                this.dialogRef.componentInstance.codesHeader = this.codesHeader;
                this.dialogRef.componentInstance.unchangeableCodes = this.codes;
                this.dialogRef.componentInstance.codes = this.codes;
                if (!this.popupWidth && Object.values(this.codes[0]).length >= 4) {
                    this.extendWidth = ShCtlConst.DIALOG_MAX_WIDTH;
                }
                /** @type {?} */
                var cols_2 = ['select'];
                this.dialogRef.componentInstance.getHashKeys().forEach((/**
                 * @param {?} name
                 * @param {?} idx
                 * @return {?}
                 */
                function (name, idx) {
                    if (!_this.hiddenColumns.find((/**
                     * @param {?} _
                     * @return {?}
                     */
                    function (_) { return _ == (idx + 1); })))
                        cols_2.push(name);
                }));
                this.dialogRef.componentInstance.displayedColumns = cols_2;
            }
        }
    };
    ShCtlPopupMultiComponent.decorators = [
        { type: Component, args: [{
                    selector: 'sh-ctl-popup-multi',
                    template: "<div *ngIf=\"!disabled\" style=\"margin-top: 17px;\">\r\n  <mat-icon mat-button matSuffix class=\"mat-search-btn\"\r\n  (click)=\"getCodes()\">search</mat-icon>\r\n</div>",
                    styles: [".input-group-text{background-color:#fff;border:none}.btn{border:none}.btn.btn-primary:disabled{background-color:#999}.btn-primary{background-color:#17a2b8}.btn-primary:hover{background-color:#1792b8}.fixed-bottom{top:708px;height:65px;max-width:1024px;margin-right:auto;margin-left:auto}.mat-search-btn{color:#bababa;font-size:1.1rem}.mat-search-btn:hover{cursor:pointer}.mat-icon-container{margin-left:-17px}.mat-icon-container .mat-search-btn{color:#bababa;font-size:1.1rem;margin:17px 0 0 100%}.result-input{text-overflow:ellipsis}.display-ctl-view{display:-webkit-box;display:flex}.display-result{margin-left:30px}.valueOutput-container{display:-webkit-box;display:flex;-webkit-box-flex:1;flex:1}.notitle-value{width:100%;margin-left:30px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
                }] }
    ];
    /** @nocollapse */
    ShCtlPopupMultiComponent.ctorParameters = function () { return [
        { type: SmplPopupService },
        { type: MatDialog }
    ]; };
    ShCtlPopupMultiComponent.propDecorators = {
        dataTitle: [{ type: Input }],
        apiUrl: [{ type: Input }],
        dataCache: [{ type: Input }],
        uniqueColumns: [{ type: Input }],
        hiddenColumns: [{ type: Input }],
        popupWidth: [{ type: Input }]
    };
    return ShCtlPopupMultiComponent;
}(SHFormControl));
export { ShCtlPopupMultiComponent };
if (false) {
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.dataTitle;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.apiUrl;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.dataCache;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.uniqueColumns;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.hiddenColumns;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.popupWidth;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.extendWidth;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.dialogRef;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.codes;
    /** @type {?} */
    ShCtlPopupMultiComponent.prototype.codesHeader;
    /**
     * @type {?}
     * @protected
     */
    ShCtlPopupMultiComponent.prototype.SmplPopupService;
    /**
     * @type {?}
     * @protected
     */
    ShCtlPopupMultiComponent.prototype.dialog;
}
