// sh-form-control/esm5/lib/components/sh-ctl-time/sh-ctl-time.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { SHCtlCommon } from '../sh-ctl-common';
import { SHFormControl } from '../../utilities/ShFormControl';
var SHCtlTimeComponent = /** @class */ (function (_super) {
    tslib_1.__extends(SHCtlTimeComponent, _super);
    function SHCtlTimeComponent() {
        var _this = _super.call(this) || this;
        _this.obj = ".ShCtlTime";
        _this.adjust = 1;
        _this.isValidateError = false;
        _this.errorMessage = "無効な日付です";
        _this.isFirst = true;
        _this.format = SHCtlCommon.Const.timeFormat.HHMM;
        //直前の値保持
        _this.timeStrage = null;
        _this.ctlName = "SHCtlTime";
        return _this;
    }
    /**
     * @return {?}
     */
    SHCtlTimeComponent.prototype.ngAfterContentChecked = /**
     * @return {?}
     */
    function () {
        if (this.value != null) {
            /** @type {?} */
            var date = this.value.split(':');
            switch (this.format) {
                case SHCtlCommon.Const.timeFormat.HH:
                    this.stringValue = "" + date[0];
                    break;
                case SHCtlCommon.Const.timeFormat.HHMM:
                    this.stringValue = date[0] + ":" + (date.length > 1 ? date[1] : "00");
                    break;
                case SHCtlCommon.Const.timeFormat.HH:
                    this.stringValue = date[0] + ":" + (date.length > 1 ? date[1] : "00") + ":" + (date.length > 2 ? date[2] : "00");
                    break;
                default:
                    break;
            }
        }
    };
    /**
     * @return {?}
     */
    SHCtlTimeComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.onInit();
    };
    /**
     * onInput
     * @param event input
     */
    /**
     * onInput
     * @param {?} event input
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onInput = /**
     * onInput
     * @param {?} event input
     * @return {?}
     */
    function (event) {
        this.timecalculation(event);
    };
    /**
     * onClick
     * @param event click
     */
    /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onClick = /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    function (event) {
        this.onBeforeClick(event);
        this.onAfterClick(event);
    };
    /**
     * onFocus
     * @param event focus
     */
    /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onFocus = /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeFocus(event);
        event.target.select();
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterFocus(event); }), 0);
    };
    /**
     * onBlur
     * @param event blur
     */
    /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onBlur = /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeBlur(event);
        /** @type {?} */
        var val = event.target.value;
        if (val) {
            //HH形式の場合
            if (val.length == 2)
                val += ":00:00";
            //HHMM形式の場合
            if (val.length == 5)
                val += ":00";
        }
        if (this.timeStrage != val) {
            this.onBeforeChange(event);
            this.valueChange.emit(val);
        }
        this.timeStrage = val;
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterBlur(event); }), 0);
    };
    /**
     * onKeypress
     * @param event keypress
     */
    /**
     * onKeypress
     * @param {?} event keypress
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onKeypress = /**
     * onKeypress
     * @param {?} event keypress
     * @return {?}
     */
    function (event) {
        this.timecalculation(event);
    };
    /**
     * onChange
     * @param event change
     */
    /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    SHCtlTimeComponent.prototype.onChange = /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    function (event) {
    };
    /**
     * キーイベント系
     * /キーを押すことで現在時刻をセット
     * @param event keypress
     */
    /**
     * キーイベント系
     * /キーを押すことで現在時刻をセット
     * @private
     * @param {?} event keypress
     * @return {?}
     */
    SHCtlTimeComponent.prototype.timecalculation = /**
     * キーイベント系
     * /キーを押すことで現在時刻をセット
     * @private
     * @param {?} event keypress
     * @return {?}
     */
    function (event) {
        if (event.key === SHCtlCommon.Const.keyCodes.signs[191]) {
            event.target.value = this.setCurrentTime();
            this.onChange(event);
        }
    };
    /**
     * 現在時刻の設定
     * @param event keypress
     */
    /**
     * 現在時刻の設定
     * @private
     * @return {?}
     */
    SHCtlTimeComponent.prototype.setCurrentTime = /**
     * 現在時刻の設定
     * @private
     * @return {?}
     */
    function () {
        /** @type {?} */
        var newDate = new Date();
        /** @type {?} */
        var nullFirstString = null;
        /** @type {?} */
        var nullLastString = null;
        nullFirstString = SHCtlCommon.padding(newDate.getHours(), 2); // HH
        nullLastString = SHCtlCommon.padding(newDate.getMinutes(), 2); // HH:MM
        // HH:MM
        /** @type {?} */
        var stringValue = String(nullFirstString + ":" + nullLastString);
        return stringValue;
    };
    SHCtlTimeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'sh-ctl-time',
                    template: "<ng-container *ngIf=\"!onlyField; else onlyFieldControl\">\r\n  <div *ngIf=\"!isDisabled\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <mat-form-field class=\"mat-validation\" [floatLabel]=\"floatLabel\">\r\n      <input #inputElm matInput [placeholder]=\"title\" [required]=\"required\" type=\"time\" step=\"{{format}}\" max=\"23:59\" [(ngModel)]=\"value\"\r\n        attr.name=\"{{ctlName}}\" class=\"form-control ShCtlTime can-moved-with-the-enter-key  {{strClass}}\"\r\n        (input)=\"onInput($event)\" (click)=\"onClick($event)\" (focus)=\"onFocus($event)\" (blur)=\"onBlur($event)\"\r\n        autocomplete=\"off\" (change)=\"onChange($event)\" [disabled]=\"disabled\" (keypress)=\"onKeypress($event)\"\r\n        [class.isValidateError]=\"isValidateError\" maxlength=\"{{maxlength}}\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" >\r\n      <button mat-button class=\"closeButton\" type=\"button\" *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n        aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n        <mat-icon>close</mat-icon>\r\n      </button>\r\n    </mat-form-field>\r\n    <span class=\"custom-alert-message\">{{message.toString().split(',').join(' ')}}</span>\r\n  </div>\r\n</ng-container>\r\n\r\n<ng-template #onlyFieldControl>\r\n  <div class=\"only-field\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <input #inputElm matInput [required]=\"required\" type=\"time\" step=\"{{format}}\" max=\"23:59\" [(ngModel)]=\"value\"\r\n      attr.name=\"{{ctlName}}\" class=\"form-control ShCtlTime can-moved-with-the-enter-key  {{strClass}}\"\r\n      (input)=\"onInput($event)\" (click)=\"onClick($event)\" (focus)=\"onFocus($event)\" (blur)=\"onBlur($event)\"\r\n      autocomplete=\"off\" (change)=\"onChange($event)\" [disabled]=\"disabled\" (keypress)=\"onKeypress($event)\"\r\n      [class.isValidateError]=\"isValidateError\" maxlength=\"{{maxlength}}\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" >\r\n    <button mat-button class=\"closeButton\" type=\"button\" *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n      aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n      <mat-icon>close</mat-icon>\r\n    </button>\r\n  </div>\r\n</ng-template>\r\n\r\n<div *ngIf=\"isDisabled\" class=\"display-ctl-view\">\r\n  <div>\r\n    <div *ngIf=\"title\" class=\"display-ctl-view\">\r\n      <div class=\"display-result\" [ngStyle]=\"{'width.px': labelWidth}\" fxLayout=\"row wrap\" fxFlex>\r\n        <div fxFlex class=\"item-title\"><strong>{{title}}:</strong></div>\r\n      </div>\r\n      <div fxFlex class=\"valueOutput\" type=\"time\">\r\n        {{stringValue}}\r\n      </div>\r\n    </div>\r\n\r\n    <div *ngIf=\"!title\" class=\"display-ctl-view notitle-value\">\r\n      <div fxFlex class=\"valueOutput\">\r\n        {{stringValue}}\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                    styles: [".isValidateError{border:1px solid red}.alert-danger{background-color:#ffd2ff}.display-ctl-view{display:-webkit-box;display:flex}.display-result{margin-left:30px}.valueOutput{-webkit-box-flex:1;flex:1;padding-top:4px}.notitle-value{width:100%;margin-left:30px}.ShCtlTime{float:left}mat-form-field .closeButton{position:absolute;right:20px;bottom:5px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
                }] }
    ];
    /** @nocollapse */
    SHCtlTimeComponent.ctorParameters = function () { return []; };
    SHCtlTimeComponent.propDecorators = {
        format: [{ type: Input }]
    };
    return SHCtlTimeComponent;
}(SHFormControl));
export { SHCtlTimeComponent };
if (false) {
    /** @type {?} */
    SHCtlTimeComponent.prototype.obj;
    /** @type {?} */
    SHCtlTimeComponent.prototype.adjust;
    /** @type {?} */
    SHCtlTimeComponent.prototype.isValidateError;
    /** @type {?} */
    SHCtlTimeComponent.prototype.errorMessage;
    /** @type {?} */
    SHCtlTimeComponent.prototype.stringValue;
    /** @type {?} */
    SHCtlTimeComponent.prototype.isFirst;
    /** @type {?} */
    SHCtlTimeComponent.prototype.format;
    /**
     * @type {?}
     * @private
     */
    SHCtlTimeComponent.prototype.timeStrage;
}
