// sh-form-control/esm5/lib/utilities/MaxByteLengthHistory.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { MaxByteLengthService } from "../services/max-bytelength.service";
var MaxByteLengthHistory = /** @class */ (function () {
    function MaxByteLengthHistory(value, startPoint, maxByteLength) {
        this.enterableBytes = 0;
        this.service = new MaxByteLengthService();
        this.cache = value;
        this.startPoint = value.length == 1 && startPoint == 1 ? 0 : startPoint;
        this.maxByteLength = maxByteLength;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    MaxByteLengthHistory.isAllowOperation = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        return MaxByteLengthService.isAllowOperation(e);
    };
    /**
     *@ enterableBytesを書き換えます
     */
    /**
     * \@ enterableBytesを書き換えます
     * @param {?} value
     * @return {?}
     */
    MaxByteLengthHistory.prototype.changeEnterableBytes = /**
     * \@ enterableBytesを書き換えます
     * @param {?} value
     * @return {?}
     */
    function (value) {
        /** @type {?} */
        var info = this.service.generateByteInfo(value);
        /** @type {?} */
        var entarableByte = this.maxByteLength - info.length;
        this.enterableBytes = entarableByte >= 0 ? entarableByte : 0;
    };
    /**
     *@ 残りの入力可能byte数を返す
     */
    /**
     * \@ 残りの入力可能byte数を返す
     * @return {?}
     */
    MaxByteLengthHistory.prototype.getEnterableBytes = /**
     * \@ 残りの入力可能byte数を返す
     * @return {?}
     */
    function () {
        return this.enterableBytes;
    };
    /**
     *@ inputにセットする値を返す
     */
    /**
     * \@ inputにセットする値を返す
     * @param {?} data
     * @param {?} value
     * @return {?}
     */
    MaxByteLengthHistory.prototype.getResult = /**
     * \@ inputにセットする値を返す
     * @param {?} data
     * @param {?} value
     * @return {?}
     */
    function (data, value) {
        if (!value)
            return "";
        // // 許容バイト数がまだある時
        if (this.enterableBytes)
            return value;
        if (this.startPoint === 0) {
            var bytes_1 = this.service.generateByteInfo(value).bytes;
            /** @type {?} */
            var maxByteLength = this.service.calcMaxLength(bytes_1, this.maxByteLength);
            return value.substr(0, maxByteLength);
        }
        var bytes = this.service.generateByteInfo(data).bytes;
        var length = this.service.generateByteInfo(this.cache).length;
        /** @type {?} */
        var len = this.service.calcMaxLength(bytes, this.maxByteLength, length);
        /** @type {?} */
        var insertStr = data.substr(0, len);
        /** @type {?} */
        var h = this.cache.substr(0, this.startPoint);
        /** @type {?} */
        var f = this.cache.substring(this.startPoint);
        return h + insertStr + f;
    };
    /**
     *アルファベットや数字の入力時に標準のmaxlengthに設定するための値を返す
     */
    /**
     * アルファベットや数字の入力時に標準のmaxlengthに設定するための値を返す
     * @param {?} value
     * @return {?}
     */
    MaxByteLengthHistory.prototype.getMaxLength = /**
     * アルファベットや数字の入力時に標準のmaxlengthに設定するための値を返す
     * @param {?} value
     * @return {?}
     */
    function (value) {
        var bytes = this.service.generateByteInfo(value).bytes;
        return this.service.calcMaxLength(bytes, this.maxByteLength);
    };
    /**
     *デバッグ用
     */
    /**
     * デバッグ用
     * @return {?}
     */
    MaxByteLengthHistory.prototype.toString = /**
     * デバッグ用
     * @return {?}
     */
    function () {
        return JSON.stringify({
            cache: this.cache,
            startPoint: this.startPoint,
            enterableBytes: this.enterableBytes,
            maxByteLength: this.maxByteLength,
        });
    };
    return MaxByteLengthHistory;
}());
export { MaxByteLengthHistory };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.cache;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.startPoint;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.enterableBytes;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.maxByteLength;
    /**
     * @type {?}
     * @private
     */
    MaxByteLengthHistory.prototype.service;
}
