// sh-http-client/esm5/lib/services/alert-handler.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { Alert } from '../utilities/Alert';
import * as i0 from "@angular/core";
var AlertHandlerService = /** @class */ (function () {
    /**
     * コンストラクタ
     */
    function AlertHandlerService() {
        this.Alert = new Alert();
        this.AlertHandlerSubject = new Subject();
        this.AlertHandlerState = this.AlertHandlerSubject.asObservable();
    }
    /**
    * アラートを追加する
    * @param validationType 検証結果タイプ
    * @param validationMessages 検証結果メッセージ配列
    */
    /**
     * アラートを追加する
     * @param {?} alertInfo
     * @return {?}
     */
    AlertHandlerService.prototype.setAlert = /**
     * アラートを追加する
     * @param {?} alertInfo
     * @return {?}
     */
    function (alertInfo) {
        this.Alert = alertInfo;
        this.AlertHandlerSubject.next(this.Alert);
    };
    /**
     * アラートをクリアする
     */
    /**
     * アラートをクリアする
     * @return {?}
     */
    AlertHandlerService.prototype.alertClear = /**
     * アラートをクリアする
     * @return {?}
     */
    function () {
        this.Alert.message = "";
        this.Alert.type = "";
        this.AlertHandlerSubject.next(this.Alert);
    };
    AlertHandlerService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    AlertHandlerService.ctorParameters = function () { return []; };
    /** @nocollapse */ AlertHandlerService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function AlertHandlerService_Factory() { return new AlertHandlerService(); }, token: AlertHandlerService, providedIn: "root" });
    return AlertHandlerService;
}());
export { AlertHandlerService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AlertHandlerService.prototype.Alert;
    /**
     * @type {?}
     * @private
     */
    AlertHandlerService.prototype.AlertHandlerSubject;
    /** @type {?} */
    AlertHandlerService.prototype.AlertHandlerState;
}
