// sh-http-client/esm5/lib/services/auth.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { ShHttpClientConst } from '../constShHttpClient';
import { ShHttpClientService } from './sh-http-client.service';
import * as i0 from "@angular/core";
import * as i1 from "./sh-http-client.service";
var AuthService = /** @class */ (function () {
    function AuthService(http) {
        this.http = http;
    }
    /**
     * 認証済みかの判定
     */
    /**
     * 認証済みかの判定
     * @return {?}
     */
    AuthService.prototype.isAuthenticated = /**
     * 認証済みかの判定
     * @return {?}
     */
    function () {
        this.token = sessionStorage.getItem("" + ShHttpClientConst.AUTHORIZATION_TOKEN);
        if (this.token) {
            return of(true);
        }
        else {
            return of(false);
        }
    };
    /**
     * 認証
     * @param jigCode
     * @param tanCode
     * @param password
     */
    /**
     * 認証
     * @param {?} jigCode
     * @param {?} tanCode
     * @param {?} password
     * @return {?}
     */
    AuthService.prototype.login = /**
     * 認証
     * @param {?} jigCode
     * @param {?} tanCode
     * @param {?} password
     * @return {?}
     */
    function (jigCode, tanCode, password) {
        //実際はデータベースなり、ユーザー管理するサービスなりに問い合わせて戻り値を返す
        //サーバーとの通信の結果をlacalStorageへ保存（今は仮実装）
        sessionStorage.setItem(ShHttpClientConst.AUTHORIZATION_TOKEN, jigCode + tanCode + password);
        //通信の結果で分岐する
        if (true) {
            return of(true);
        }
        else {
            /**認証失敗時 */
            return of(false);
        }
    };
    /**
     * ログアウト
     * リダイレクトはガードで行う
     */
    /**
     * ログアウト
     * リダイレクトはガードで行う
     * @param {?} url
     * @return {?}
     */
    AuthService.prototype.logout = /**
     * ログアウト
     * リダイレクトはガードで行う
     * @param {?} url
     * @return {?}
     */
    function (url) {
        return this.http.get(url).then((/**
         * @param {?} _
         * @return {?}
         */
        function (_) {
            sessionStorage.removeItem(ShHttpClientConst.AUTHORIZATION_TOKEN);
        }));
    };
    AuthService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    AuthService.ctorParameters = function () { return [
        { type: ShHttpClientService }
    ]; };
    /** @nocollapse */ AuthService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function AuthService_Factory() { return new AuthService(i0.ɵɵinject(i1.ShHttpClientService)); }, token: AuthService, providedIn: "root" });
    return AuthService;
}());
export { AuthService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AuthService.prototype.token;
    /**
     * @type {?}
     * @private
     */
    AuthService.prototype.http;
}
