// sh-http-client/esm5/lib/services/logger.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShHttpClientService } from './sh-http-client.service';
import { Router, NavigationEnd } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "./sh-http-client.service";
import * as i2 from "@angular/router";
var LoggerService = /** @class */ (function () {
    function LoggerService(http, router) {
        this.http = http;
        this.router = router;
        this.isBackButton = false;
        this.config = { url: '', useHash: true };
    }
    /**
     * logApiへリクエスト
     *
     */
    /**
     * logApiへリクエスト
     *
     * @private
     * @param {?} param
     * @return {?}
     */
    LoggerService.prototype.push = /**
     * logApiへリクエスト
     *
     * @private
     * @param {?} param
     * @return {?}
     */
    function (param) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                return [2 /*return*/, this.http.post(this.config.url, param)];
            });
        });
    };
    /**
     * 設定を更新
     */
    /**
     * 設定を更新
     * @template THIS
     * @this {THIS}
     * @param {?} config
     * @return {THIS}
     */
    LoggerService.prototype.configure = /**
     * 設定を更新
     * @template THIS
     * @this {THIS}
     * @param {?} config
     * @return {THIS}
     */
    function (config) {
        var url = config.url, useHash = config.useHash;
        if (url) {
            (/** @type {?} */ (this)).config.url = url;
        }
        if (useHash !== undefined && useHash !== null) {
            (/** @type {?} */ (this)).config.useHash = useHash;
        }
        return (/** @type {?} */ (this));
    };
    /**
     * 画面遷移が発生するたびにlogging
     */
    /**
     * 画面遷移が発生するたびにlogging
     * @return {?}
     */
    LoggerService.prototype.trackingPath = /**
     * 画面遷移が発生するたびにlogging
     * @return {?}
     */
    function () {
        var _this = this;
        this.router.events.subscribe((/**
         * @param {?} event
         * @return {?}
         */
        function (event) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var path, result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (!(event instanceof NavigationEnd)) return [3 /*break*/, 2];
                        if (this.isBackButton) {
                            this.isBackButton = false;
                            return [2 /*return*/];
                        }
                        path = this.config.useHash
                            ? "#" + event.urlAfterRedirects
                            : event.urlAfterRedirects.replace('/', '');
                        return [4 /*yield*/, this.push({ path: path })];
                    case 1:
                        result = _a.sent();
                        _a.label = 2;
                    case 2: return [2 /*return*/];
                }
            });
        }); }));
    };
    /**
     * actionlog pathは必須のため必ず送られる
     */
    /**
     * actionlog pathは必須のため必ず送られる
     * @param {?} action
     * @return {?}
     */
    LoggerService.prototype.action = /**
     * actionlog pathは必須のため必ず送られる
     * @param {?} action
     * @return {?}
     */
    function (action) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = this.config.useHash
                            ? "#" + this.router.url
                            : this.router.url.replace('/', '');
                        return [4 /*yield*/, this.push({ path: path, action: action })];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     *
     */
    /**
     *
     * @return {?}
     */
    LoggerService.prototype.stopLoggingPath = /**
     *
     * @return {?}
     */
    function () {
        this.isBackButton = true;
    };
    LoggerService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    LoggerService.ctorParameters = function () { return [
        { type: ShHttpClientService },
        { type: Router }
    ]; };
    /** @nocollapse */ LoggerService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function LoggerService_Factory() { return new LoggerService(i0.ɵɵinject(i1.ShHttpClientService), i0.ɵɵinject(i2.Router)); }, token: LoggerService, providedIn: "root" });
    return LoggerService;
}());
export { LoggerService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.isBackButton;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.config;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.http;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.router;
}
