// sh-http-client/esm5/lib/services/navigation.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
var NavigationService = /** @class */ (function () {
    /**
     * コンストラクタ
     */
    function NavigationService() {
        //ユーザーのブラウザバック実行状態
        this.isBackClicked = false;
    }
    /**
     * ユーザーのブラウザバック実行を保存する
     * @return true/false
     */
    /**
     * ユーザーのブラウザバック実行を保存する
     * @return {?} true/false
     */
    NavigationService.prototype.getBackClicked = /**
     * ユーザーのブラウザバック実行を保存する
     * @return {?} true/false
     */
    function () {
        return this.isBackClicked;
    };
    /**
     * ユーザーのブラウザバック実行を保存する
     * @param isBackClicked
     */
    /**
     * ユーザーのブラウザバック実行を保存する
     * @param {?} isBackClicked
     * @return {?}
     */
    NavigationService.prototype.setBackClicked = /**
     * ユーザーのブラウザバック実行を保存する
     * @param {?} isBackClicked
     * @return {?}
     */
    function (isBackClicked) {
        this.isBackClicked = isBackClicked;
    };
    NavigationService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NavigationService.ctorParameters = function () { return []; };
    /** @nocollapse */ NavigationService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NavigationService_Factory() { return new NavigationService(); }, token: NavigationService, providedIn: "root" });
    return NavigationService;
}());
export { NavigationService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NavigationService.prototype.isBackClicked;
}
