// sh-http-client/esm5/lib/services/sh-redirect.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, InjectionToken, Inject } from '@angular/core';
import { Router } from '@angular/router';
import { ShHttpClientConst } from '../constShHttpClient';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/** @type {?} */
export var SH_BROWSER_STORAGE = new InjectionToken("Browser Storage", {
    providedIn: "root",
    factory: (/**
     * @return {?}
     */
    function () { return localStorage; })
});
/** @type {?} */
export var USE_HASH = new InjectionToken('UseHash', {
    providedIn: "root",
    factory: (/**
     * @return {?}
     */
    function () { return ({ useHash: true }); })
});
var ShRedirectService = /** @class */ (function () {
    function ShRedirectService(router, storage, isHash) {
        this.router = router;
        this.storage = storage;
        this.isHash = isHash;
    }
    Object.defineProperty(ShRedirectService.prototype, "redirectPath", {
        get: /**
         * @return {?}
         */
        function () {
            return this.storage.getItem(ShHttpClientConst.REDIRECT_PATH);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ShRedirectService.prototype, "queryParams", {
        get: /**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var query = new URLSearchParams(this.storage.getItem(ShHttpClientConst.SH_QUERY_PARAMS)) || null;
            return query;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    ShRedirectService.prototype.setRedirectPath = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var path = '';
        /** @type {?} */
        var query = '';
        if (this.isHash.useHash) {
            /** @type {?} */
            var pathparams = location.href.split('#')[1].split('?');
            path = pathparams[0];
            query = pathparams[1] || '';
        }
        else {
            path = location.pathname;
            path = location.search;
        }
        this.storage.setItem(ShHttpClientConst.REDIRECT_PATH, path);
        this.storage.setItem(ShHttpClientConst.SH_QUERY_PARAMS, query);
    };
    /**
     * @return {?}
     */
    ShRedirectService.prototype.redirect = /**
     * @return {?}
     */
    function () {
        this.router.navigate([this.storage.getItem(ShHttpClientConst.REDIRECT_PATH)]);
        this.storage.setItem(ShHttpClientConst.REDIRECT_PATH, '');
    };
    ShRedirectService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ShRedirectService.ctorParameters = function () { return [
        { type: Router },
        { type: Storage, decorators: [{ type: Inject, args: [SH_BROWSER_STORAGE,] }] },
        { type: undefined, decorators: [{ type: Inject, args: [USE_HASH,] }] }
    ]; };
    /** @nocollapse */ ShRedirectService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ShRedirectService_Factory() { return new ShRedirectService(i0.ɵɵinject(i1.Router), i0.ɵɵinject(SH_BROWSER_STORAGE), i0.ɵɵinject(USE_HASH)); }, token: ShRedirectService, providedIn: "root" });
    return ShRedirectService;
}());
export { ShRedirectService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.router;
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.storage;
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.isHash;
}
