// sh-form-control/esm2015/lib/directives/sh-ctl-state.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, Input } from '@angular/core';
export class ShCtlStateDirective {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        this.state = "";
        this.onlyField = false;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.addShCtlStateClass();
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        this.addShCtlStateClass();
    }
    /**
     * @return {?}
     */
    addShCtlStateClass() {
        /** @type {?} */
        let qs = this.el.nativeElement.querySelectorAll('mat-form-field');
        for (var e of qs) {
            /** @type {?} */
            let classList = (e) ? e.className.split(' ') : [];
            classList.forEach((/**
             * @param {?} className
             * @return {?}
             */
            className => {
                if (className.startsWith("sh-ctl-state-"))
                    e.classList.remove(className);
            }));
            e.classList.add("sh-ctl-state-" + this.state);
        }
        if (this.onlyField) {
            /** @type {?} */
            let qsOnlyField = this.el.nativeElement.querySelectorAll('.only-field');
            for (var e of qsOnlyField) {
                /** @type {?} */
                let classList = (e) ? e.className.split(' ') : [];
                classList.forEach((/**
                 * @param {?} className
                 * @return {?}
                 */
                className => {
                    if (className.startsWith("only-field-")) {
                        e.classList.remove(className);
                    }
                }));
                e.classList.add("only-field-" + this.state);
            }
        }
        qs = this.el.nativeElement.querySelectorAll('span,label');
        for (var e of qs) {
            /** @type {?} */
            let classList = (e) ? e.className.split(' ') : [];
            classList.forEach((/**
             * @param {?} className
             * @return {?}
             */
            className => {
                if (className.startsWith("sh-ctl-state-"))
                    e.classList.remove(className);
            }));
            e.classList.add("sh-ctl-state-" + this.state + "-message");
        }
    }
}
ShCtlStateDirective.decorators = [
    { type: Directive, args: [{
                selector: '[state]'
            },] }
];
/** @nocollapse */
ShCtlStateDirective.ctorParameters = () => [
    { type: ElementRef }
];
ShCtlStateDirective.propDecorators = {
    state: [{ type: Input, args: ['state',] }],
    onlyField: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    ShCtlStateDirective.prototype.state;
    /** @type {?} */
    ShCtlStateDirective.prototype.onlyField;
    /**
     * @type {?}
     * @private
     */
    ShCtlStateDirective.prototype.el;
}
