// sh-form-control/esm2015/lib/directives/sh-readonly.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, Input } from '@angular/core';
export class ShReadOnlyDirective {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        this.isReadOnly = true;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.addAttribute();
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        this.addAttribute();
    }
    /**
     * @return {?}
     */
    addAttribute() {
        /** @type {?} */
        let qs = this.el.nativeElement.querySelectorAll('input,textarea,select');
        for (var e of qs) {
            e.readOnly = this.isReadOnly;
        }
        qs = this.el.nativeElement.querySelectorAll('button,mat-icon');
        if (this.isReadOnly) {
            for (var e of qs) {
                e.style.display = "none";
            }
        }
        else {
            for (var e of qs) {
                e.style.display = "inline";
            }
        }
    }
}
ShReadOnlyDirective.decorators = [
    { type: Directive, args: [{
                selector: '[readonly]'
            },] }
];
/** @nocollapse */
ShReadOnlyDirective.ctorParameters = () => [
    { type: ElementRef }
];
ShReadOnlyDirective.propDecorators = {
    isReadOnly: [{ type: Input, args: ['readonly',] }]
};
if (false) {
    /** @type {?} */
    ShReadOnlyDirective.prototype.isReadOnly;
    /**
     * @type {?}
     * @private
     */
    ShReadOnlyDirective.prototype.el;
}
