// sh-form-control/esm5/lib/components/sh-ctl-lock-screen/sh-ctl-lock-screen.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { LockScreenService } from '../../services/lock-screen.service';
var SHCtlLockScreenComponent = /** @class */ (function () {
    /**
    * コンストラクタ
    */
    function SHCtlLockScreenComponent(lockScreen) {
        this.lockScreen = lockScreen;
        this.isConnecting = false;
    }
    /**
    * コンポーネント初期化
    */
    /**
     * コンポーネント初期化
     * @return {?}
     */
    SHCtlLockScreenComponent.prototype.ngOnInit = /**
     * コンポーネント初期化
     * @return {?}
     */
    function () {
        var _this = this;
        this.lockScreen.isConnectingState.subscribe((/**
         * @param {?} isConnecting
         * @return {?}
         */
        function (isConnecting) {
            _this.isConnecting = isConnecting;
        }));
    };
    SHCtlLockScreenComponent.decorators = [
        { type: Component, args: [{
                    selector: 'sh-ctl-lock-screen',
                    template: "<div *ngIf=\"isConnecting\" class=\"overlay\">\r\n    <mat-spinner></mat-spinner>\r\n    <span class=\"loader-text\">Loading...</span>\r\n</div>\r\n",
                    styles: [".overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}"]
                }] }
    ];
    /** @nocollapse */
    SHCtlLockScreenComponent.ctorParameters = function () { return [
        { type: LockScreenService }
    ]; };
    return SHCtlLockScreenComponent;
}());
export { SHCtlLockScreenComponent };
if (false) {
    /** @type {?} */
    SHCtlLockScreenComponent.prototype.isConnecting;
    /**
     * @type {?}
     * @private
     */
    SHCtlLockScreenComponent.prototype.lockScreen;
}
