// sh-form-control/esm5/lib/components/sh-ctl-kbn-radio/sh-ctl-kbn-radio.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Input, Output } from "@angular/core";
import { SmplKbnCodeService } from "../../services/smpl-kbn-code.service";
import { SHFormControl } from '../../utilities/ShFormControl';
var SHCtlKbnRadioComponent = /** @class */ (function (_super) {
    tslib_1.__extends(SHCtlKbnRadioComponent, _super);
    function SHCtlKbnRadioComponent(SmplKbnRadioService) {
        var _this = _super.call(this) || this;
        _this.SmplKbnRadioService = SmplKbnRadioService;
        _this.dataKbnName = "";
        _this.radioContent = "";
        _this.apiUrl = "/api/";
        _this.displayLabel = "";
        _this.throwError = new EventEmitter();
        _this.name = "";
        _this.ctlName = "SHCtlKbnRadio";
        return _this;
    }
    /**
     *プロパティの初期化
     *通常サービスにキャッシュしているが、無ければ再度マスタより取得
     */
    /**
     * プロパティの初期化
     * 通常サービスにキャッシュしているが、無ければ再度マスタより取得
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.ngOnInit = /**
     * プロパティの初期化
     * 通常サービスにキャッシュしているが、無ければ再度マスタより取得
     * @return {?}
     */
    function () {
        var _this = this;
        this.onInit();
        if (!this.items) {
            if (this.SmplKbnRadioService.getMasterData(this.apiUrl).length == 0) {
                this.SmplKbnRadioService.fetchAll(this.apiUrl).then((/**
                 * @param {?} response
                 * @return {?}
                 */
                function (response) {
                    _this.kbnCodes = response;
                    _this.SmplKbnRadioService.setMasterData(_this.apiUrl, response);
                }));
            }
            else {
                if (this.SmplKbnRadioService.getMasterData(this.apiUrl).length > 0)
                    this.kbnCodes = this.SmplKbnRadioService.getMasterData(this.apiUrl);
            }
        }
        else {
            this.kbnCodes = this.items;
        }
    };
    /**
     * ngOnChanges
     */
    /**
     * ngOnChanges
     * @param {?} change
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.ngOnChanges = /**
     * ngOnChanges
     * @param {?} change
     * @return {?}
     */
    function (change) {
        if (change.items && change.items.currentValue) {
            this.kbnCodes = change.items.currentValue;
        }
        if (change.value) {
            if (change.value.currentValue) {
                this.value = change.value.currentValue.toString();
            }
        }
    };
    /**
     * onClick
     * @param event click
     */
    /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.onClick = /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    function (event) {
        this.onBeforeClick(event);
        this.onAfterClick(event);
    };
    /**
     * onChange
     * @param event change
     */
    /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.onChange = /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeChange(event);
        this.valueChange.emit(event.value);
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterChange(event); }), 0);
    };
    /**
     * onFocus
     * @param event focus
     */
    /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.onFocus = /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeFocus(event);
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterFocus(event); }), 0);
    };
    /**
     * onBlur
     * @param event blur
     */
    /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.onBlur = /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeBlur(event);
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterBlur(event); }), 0);
        ;
    };
    /**
     * フォーマット編集
     * @param value this.value
     */
    /**
     * フォーマット編集
     * @param {?} item
     * @return {?}
     */
    SHCtlKbnRadioComponent.prototype.changeRadioContent = /**
     * フォーマット編集
     * @param {?} item
     * @return {?}
     */
    function (item) {
        this.radioContent = item;
    };
    SHCtlKbnRadioComponent.decorators = [
        { type: Component, args: [{
                    selector: "sh-ctl-kbn-radio",
                    template: "<ng-container *ngIf=\"!onlyField; else onlyFieldControl\">\r\n  <div *ngIf=\"!isDisabled\" style=\"height: 52.5px;\">\r\n    <div style=\"height: 52.5px;\">\r\n      <label class=\"sh-label\">{{title}}</label>\r\n      <mat-radio-group [(ngModel)]=\"value\" [required]=\"required\" aria-label=\"Select an option\" [disabled]=\"disabled\">\r\n        <mat-radio-button *ngFor=\"let item of kbnCodes\" value=\"{{item.key}}\" attr.name=\"{{ctlName}}\"\r\n           (click)=\"onClick($event)\" (change)=\"onChange($event)\" (focus)=\"onFocus($event)\"\r\n          (blur)=\"onBlur($event)\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" class=\"{{strClass}}\">\r\n          {{item.value}}</mat-radio-button>\r\n      </mat-radio-group>\r\n    </div>\r\n    <span class=\"custom-alert-message\" style=\"top:0px\">{{message.toString().split(',').join(' ')}}</span>\r\n  </div>\r\n</ng-container>\r\n\r\n<ng-template #onlyFieldControl>\r\n  <div class=\"only-field\">\r\n    <div>\r\n      <mat-radio-group [(ngModel)]=\"value\" [required]=\"required\" aria-label=\"Select an option\" [disabled]=\"disabled\">\r\n        <mat-radio-button *ngFor=\"let item of kbnCodes\" value=\"{{item.key}}\" attr.name=\"{{ctlName}}\"\r\n           (click)=\"onClick($event)\" (change)=\"onChange($event)\" (focus)=\"onFocus($event)\"\r\n          (blur)=\"onBlur($event)\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" class=\"{{strClass}}\">\r\n          {{item.value}}</mat-radio-button>\r\n      </mat-radio-group>\r\n    </div>\r\n  </div>\r\n</ng-template>\r\n\r\n<div *ngIf=\"isDisabled\" class=\"display-ctl-view\">\r\n  <div>\r\n    <div *ngIf=\"title\" class=\"display-ctl-view\">\r\n      <div class=\"display-result\" [ngStyle]=\"{'width.px': labelWidth}\" fxLayout=\"row wrap\" fxFlex>\r\n        <div fxFlex class=\"item-title\"><strong>{{title}}:</strong></div>\r\n      </div>\r\n      <div *ngFor=\"let item of kbnCodes;\">\r\n        <div fxFlex class=\"valueOutput\" *ngIf=\"value == item.key\">\r\n          {{item.value}}\r\n        </div>\r\n      </div>\r\n    </div>\r\n\r\n    <div *ngIf=\"!title\" class=\"display-ctl-view notitle-value\">\r\n      <div *ngFor=\"let item of kbnCodes;\">\r\n        <div fxFlex class=\"valueOutput\" *ngIf=\"value == item.key\">\r\n          {{item.value}}\r\n        </div>\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                    styles: [".sh-label{font-size:.75rem;min-height:18px}.sh-radio{margin:0 10px}.display-ctl-view{display:-webkit-box;display:flex}.display-result{margin-left:30px}.valueOutput{-webkit-box-flex:1;flex:1}.mat-radio-button~.mat-radio-button{margin-left:16px}.notitle-value{width:100%;margin-left:30px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
                }] }
    ];
    /** @nocollapse */
    SHCtlKbnRadioComponent.ctorParameters = function () { return [
        { type: SmplKbnCodeService }
    ]; };
    SHCtlKbnRadioComponent.propDecorators = {
        dataKbnName: [{ type: Input }],
        radioContent: [{ type: Input }],
        apiUrl: [{ type: Input }],
        displayLabel: [{ type: Input }],
        throwError: [{ type: Output }]
    };
    return SHCtlKbnRadioComponent;
}(SHFormControl));
export { SHCtlKbnRadioComponent };
if (false) {
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.dataKbnName;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.radioContent;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.apiUrl;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.displayLabel;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.throwError;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.name;
    /** @type {?} */
    SHCtlKbnRadioComponent.prototype.kbnCodes;
    /**
     * @type {?}
     * @private
     */
    SHCtlKbnRadioComponent.prototype.SmplKbnRadioService;
}
