// sh-form-control/esm2015/lib/components/sh-ctl-jig-code/sh-ctl-jig-code.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { SHCtlCommon } from '../sh-ctl-common';
import { MatDialog } from '@angular/material';
import { SHFormPopup } from '../../utilities/ShFormPopup';
import { SmplPopupService } from '../../services/smpl-popup.service';
import { BindCtlService } from '../../services/bind-ctl.service';
import { LockScreenService } from "../../services/lock-screen.service";
export class SHCtlJigCodeComponent extends SHFormPopup {
    /**
     * @param {?} smplPopupService
     * @param {?} dialog
     * @param {?} bindCtlService
     * @param {?} lockScreenService
     */
    constructor(smplPopupService, dialog, bindCtlService, lockScreenService) {
        super(smplPopupService, dialog, bindCtlService, lockScreenService);
        this.smplPopupService = smplPopupService;
        this.dialog = dialog;
        this.bindCtlService = bindCtlService;
        this.lockScreenService = lockScreenService;
        this.ctlName = "SHCtlJigCode";
        this.maxlength = 6;
    }
    /**
     * onInput
     * @param {?} event input
     * @return {?}
     */
    onInput(event) {
        SHCtlCommon.replaceSpace(event);
        SHCtlCommon.replaceFullWidth(event);
        SHCtlCommon.replaceHalfWidth(event);
        SHCtlCommon.replaceSign(event);
        SHCtlCommon.sliceLength(event, this.maxlength);
    }
    /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    onFocus(event) {
        this.onBeforeFocus(event);
        event.target.value = this.toText(event.target.value);
        event.target.select();
        setTimeout((/**
         * @return {?}
         */
        () => { this.onAfterFocus(event); }), 0);
    }
    /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    onBlur(event) {
        this.onBeforeBlur(event);
        setTimeout((/**
         * @return {?}
         */
        () => { this.onAfterBlur(event); }), 0);
        ;
    }
    /**
     * 3桁入力時に残りを補完する
     * @param {?} jigCode 事業所コード
     * @return {?}
     */
    toText(jigCode) {
        // ブランクなら
        if (SHCtlCommon.isBlank(jigCode))
            return "";
        // 6桁は処理しない
        if (jigCode.trim().length == 6)
            return jigCode;
        /** @type {?} */
        let result = null;
        if (jigCode.trim().length == 3) {
            /** @type {?} */
            let firstString = parseInt((Number(jigCode.substr(0, 1)) * 9).toString(), 10);
            // 一文字目
            /** @type {?} */
            let secondString = parseInt((Number(jigCode.substr(1, 1)) * 8).toString(), 10);
            // 弐文字目
            /** @type {?} */
            let thirdString = parseInt((Number(jigCode.substr(2, 1)) * 7).toString(), 10);
            // 参文字目
            /** @type {?} */
            const strNo = firstString + secondString + thirdString;
            /** @type {?} */
            const lastStrting = strNo.toString().substr(strNo.toString().length - 1, 1);
            // one + two + three + '00' + lastNo
            result = `${jigCode}00${lastStrting}`;
        }
        else {
            result = "";
        }
        return result;
    }
}
SHCtlJigCodeComponent.decorators = [
    { type: Component, args: [{
                selector: 'sh-ctl-jig-code',
                template: "<ng-container *ngIf=\"!onlyField; else onlyFieldControl\">\r\n  <div *ngIf=\"!isDisabled\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <input class=\"bind-flg__input\" [(value)]=\"noChangeFlg\" [name]=\"noChangeFlgCtlName\" (change)=\"noChangeFlgChange($event)\"/>\r\n    <div class=\"sh-flex sh-flex-center\">\r\n      <mat-form-field style=\"width: 33.3%;\" class=\"mat-validation\" [floatLabel]=\"floatLabel\">\r\n        <input [required]=\"required\" #inputElm matInput (input)=\"onInput($event)\"\r\n          (click)=\"onClick($event)\" (focus)=\"onFocus($event)\" (change)=\"onChange($event)\" [(ngModel)]=\"value\"\r\n          class=\"SHCtlJigCode can-moved-with-the-enter-key {{strClass}}\" [disabled]=\"disabled\" type=\"tel\"\r\n          [placeholder]=\"title\" [ngStyle]=\"styles\" [ngClass]=\"objClass\" attr.name=\"{{ctlName}}\" />\r\n        <button mat-button *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n          aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n          <mat-icon>close</mat-icon>\r\n        </button>\r\n      </mat-form-field>\r\n      <mat-form-field class=\"result-container\" style=\"width: 66.6%;\">\r\n        <input class=\"result-input {{strClass}}\" matInput type=\"text\" [(ngModel)]=\"name\" [ngStyle]=\"styles\"\r\n        attr.name=\"{{popupLabelName}}\" [ngClass]=\"objClass\" disabled>\r\n        <mat-icon mat-button matSuffix class=\"mat-search-btn\" *ngIf=\"!inputElm.readOnly && !inputElm.disabled\"\r\n          (click)=\"getCodes()\">search</mat-icon>\r\n      </mat-form-field>\r\n    </div>\r\n    <span class=\"custom-alert-message\">{{message.toString().split(',').join(' ')}}\r\n    </span>\r\n  </div>\r\n</ng-container>\r\n\r\n<ng-template #onlyFieldControl>\r\n  <div class=\"sh-flex sh-flex-center\" (focusin)=\"onFocusin($event)\" (focusout)=\"onFocusout($event)\">\r\n    <div class=\"only-field\" style=\"width: 33.3%;\">\r\n      <input class=\"bind-flg__input\" [(value)]=\"noChangeFlg\" [name]=\"noChangeFlgCtlName\" (change)=\"noChangeFlgChange($event)\"/>\r\n      <input [required]=\"required\" #inputElm matInput (input)=\"onInput($event)\"\r\n        (click)=\"onClick($event)\" (focus)=\"onFocus($event)\" (change)=\"onChange($event)\" [(ngModel)]=\"value\"\r\n        class=\"SHCtlJigCode can-moved-with-the-enter-key {{strClass}}\" [disabled]=\"disabled\" type=\"tel\"\r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\" attr.name=\"{{ctlName}}\" />\r\n      <button mat-button *ngIf=\"inputElm.value && !inputElm.readOnly && !inputElm.disabled\" matSuffix mat-icon-button\r\n        aria-label=\"Clear\" (mousedown)=\"clearValue()\" (keydown)=\"clickClearButton($event)\">\r\n        <mat-icon>close</mat-icon>\r\n      </button>\r\n    </div>\r\n    <div class=\"only-field only-label\" style=\"width: 66.6%;\">\r\n      <input class=\"result-input {{strClass}}\" matInput type=\"text\" [(ngModel)]=\"name\" [ngStyle]=\"styles\"\r\n      attr.name=\"{{popupLabelName}}\" [ngClass]=\"objClass\" disabled>\r\n      <mat-icon mat-button matSuffix class=\"mat-search-btn\" *ngIf=\"!inputElm.readOnly && !inputElm.disabled\"\r\n        (click)=\"getCodes()\">search</mat-icon>\r\n    </div>\r\n  </div>\r\n</ng-template>\r\n\r\n<div *ngIf=\"isDisabled\" class=\"display-ctl-view\">\r\n  <div>\r\n    <div *ngIf=\"title\" class=\"display-ctl-view\">\r\n      <div class=\"display-result\" [ngStyle]=\"{'width.px': labelWidth}\" fxLayout=\"row wrap\" fxFlex>\r\n        <div fxFlex class=\"item-title\"><strong>{{title}}:</strong></div>\r\n      </div>\r\n      <div class=\"valueOutput-container\">\r\n        <div fxFlex class=\"valueOutput\">{{value}}</div>\r\n        <div fxFlex class=\"valueOutput-second\">{{name}}</div>\r\n      </div>\r\n    </div>\r\n\r\n    <div *ngIf=\"!title\" class=\"display-ctl-view notitle-value\">\r\n      <div class=\"valueOutput-container\">\r\n        <div fxFlex class=\"valueOutput\">{{value}}</div>\r\n        <div fxFlex class=\"valueOutput-second\">{{name}}</div>\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                styles: [".mat-search-btn{color:#bababa;font-size:1.1rem}.mat-search-btn:hover{cursor:pointer}.result-input{text-overflow:ellipsis;padding-right:10%}.display-ctl-view{display:-webkit-box;display:flex}.display-block{display:inline-block}.display-result{margin-left:30px}.valueOutput-container{display:-webkit-box;display:flex;-webkit-box-flex:1;flex:1}.valueOutput{margin-right:14px;padding-top:4px}.validation-error::ng-deep .mat-form-field-label{color:red}.validation-error::ng-deep .mat-form-field-underline{background-color:red}.notitle-value{width:100%;margin-left:30px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
            }] }
];
/** @nocollapse */
SHCtlJigCodeComponent.ctorParameters = () => [
    { type: SmplPopupService },
    { type: MatDialog },
    { type: BindCtlService },
    { type: LockScreenService }
];
if (false) {
    /**
     * @type {?}
     * @protected
     */
    SHCtlJigCodeComponent.prototype.smplPopupService;
    /**
     * @type {?}
     * @protected
     */
    SHCtlJigCodeComponent.prototype.dialog;
    /**
     * @type {?}
     * @protected
     */
    SHCtlJigCodeComponent.prototype.bindCtlService;
    /**
     * @type {?}
     * @protected
     */
    SHCtlJigCodeComponent.prototype.lockScreenService;
}
