// sh-form-control/esm5/lib/components/sh-ctl-validated-error/sh-ctl-validated-error.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { ValidationErrorService } from 'sh-http-client';
var ShCtlValidatedErrorComponent = /** @class */ (function () {
    /**
     * コンストラクタ
     */
    function ShCtlValidatedErrorComponent(validationErrorService) {
        this.validationErrorService = validationErrorService;
        this.isError = false;
    }
    /**
     * コンポーネント初期化
     */
    /**
     * コンポーネント初期化
     * @return {?}
     */
    ShCtlValidatedErrorComponent.prototype.ngOnInit = /**
     * コンポーネント初期化
     * @return {?}
     */
    function () {
        var _this = this;
        this.subscription = this.validationErrorService.validationErrorState
            .subscribe((/**
         * @param {?} response
         * @return {?}
         */
        function (response) {
            if (response.show) {
                _this.isError = true;
                _this.errors = [];
                Object.keys(response.meesages).forEach((/**
                 * @param {?} controlName
                 * @return {?}
                 */
                function (controlName) {
                    _this.errors = tslib_1.__spread(_this.errors, response.meesages[controlName]);
                }));
            }
            else {
                _this.isError = false;
                _this.errors = [];
            }
        }));
    };
    /**
     * @return {?}
     */
    ShCtlValidatedErrorComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.subscription.unsubscribe();
    };
    ShCtlValidatedErrorComponent.decorators = [
        { type: Component, args: [{
                    selector: 'sh-ctl-validated-error',
                    template: "<div *ngIf=\"isError\" class=\"error-message\">\r\n  <div *ngFor=\"let item of errors\" class=\"errorMessages text-red\" role=\"alert\">\r\n    \u30FB{{item}}\r\n  </div>\r\n</div>\r\n",
                    styles: [".error-message{text-align:left;width:100%;height:100%}.text-red{color:red}"]
                }] }
    ];
    /** @nocollapse */
    ShCtlValidatedErrorComponent.ctorParameters = function () { return [
        { type: ValidationErrorService }
    ]; };
    ShCtlValidatedErrorComponent.propDecorators = {
        message: [{ type: Input }],
        isError: [{ type: Input }],
        errors: [{ type: Input }]
    };
    return ShCtlValidatedErrorComponent;
}());
export { ShCtlValidatedErrorComponent };
if (false) {
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.message;
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.isError;
    /** @type {?} */
    ShCtlValidatedErrorComponent.prototype.errors;
    /**
     * @type {?}
     * @private
     */
    ShCtlValidatedErrorComponent.prototype.subscription;
    /**
     * @type {?}
     * @private
     */
    ShCtlValidatedErrorComponent.prototype.validationErrorService;
}
