// sh-form-control/esm5/lib/directives/sh-ctl-state.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Directive, ElementRef, Input } from '@angular/core';
var ShCtlStateDirective = /** @class */ (function () {
    function ShCtlStateDirective(el) {
        this.el = el;
        this.state = "";
        this.onlyField = false;
    }
    /**
     * @return {?}
     */
    ShCtlStateDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.addShCtlStateClass();
    };
    /**
     * @return {?}
     */
    ShCtlStateDirective.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.addShCtlStateClass();
    };
    /**
     * @return {?}
     */
    ShCtlStateDirective.prototype.addShCtlStateClass = /**
     * @return {?}
     */
    function () {
        var e_1, _a, e_2, _b, e_3, _c;
        /** @type {?} */
        var qs = this.el.nativeElement.querySelectorAll('mat-form-field');
        try {
            for (var qs_1 = tslib_1.__values(qs), qs_1_1 = qs_1.next(); !qs_1_1.done; qs_1_1 = qs_1.next()) {
                var e = qs_1_1.value;
                /** @type {?} */
                var classList = (e) ? e.className.split(' ') : [];
                classList.forEach((/**
                 * @param {?} className
                 * @return {?}
                 */
                function (className) {
                    if (className.startsWith("sh-ctl-state-"))
                        e.classList.remove(className);
                }));
                e.classList.add("sh-ctl-state-" + this.state);
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (qs_1_1 && !qs_1_1.done && (_a = qs_1.return)) _a.call(qs_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
        if (this.onlyField) {
            /** @type {?} */
            var qsOnlyField = this.el.nativeElement.querySelectorAll('.only-field');
            try {
                for (var qsOnlyField_1 = tslib_1.__values(qsOnlyField), qsOnlyField_1_1 = qsOnlyField_1.next(); !qsOnlyField_1_1.done; qsOnlyField_1_1 = qsOnlyField_1.next()) {
                    var e = qsOnlyField_1_1.value;
                    /** @type {?} */
                    var classList = (e) ? e.className.split(' ') : [];
                    classList.forEach((/**
                     * @param {?} className
                     * @return {?}
                     */
                    function (className) {
                        if (className.startsWith("only-field-")) {
                            e.classList.remove(className);
                        }
                    }));
                    e.classList.add("only-field-" + this.state);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (qsOnlyField_1_1 && !qsOnlyField_1_1.done && (_b = qsOnlyField_1.return)) _b.call(qsOnlyField_1);
                }
                finally { if (e_2) throw e_2.error; }
            }
        }
        qs = this.el.nativeElement.querySelectorAll('span,label');
        try {
            for (var qs_2 = tslib_1.__values(qs), qs_2_1 = qs_2.next(); !qs_2_1.done; qs_2_1 = qs_2.next()) {
                var e = qs_2_1.value;
                /** @type {?} */
                var classList = (e) ? e.className.split(' ') : [];
                classList.forEach((/**
                 * @param {?} className
                 * @return {?}
                 */
                function (className) {
                    if (className.startsWith("sh-ctl-state-"))
                        e.classList.remove(className);
                }));
                e.classList.add("sh-ctl-state-" + this.state + "-message");
            }
        }
        catch (e_3_1) { e_3 = { error: e_3_1 }; }
        finally {
            try {
                if (qs_2_1 && !qs_2_1.done && (_c = qs_2.return)) _c.call(qs_2);
            }
            finally { if (e_3) throw e_3.error; }
        }
    };
    ShCtlStateDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[state]'
                },] }
    ];
    /** @nocollapse */
    ShCtlStateDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ShCtlStateDirective.propDecorators = {
        state: [{ type: Input, args: ['state',] }],
        onlyField: [{ type: Input }]
    };
    return ShCtlStateDirective;
}());
export { ShCtlStateDirective };
if (false) {
    /** @type {?} */
    ShCtlStateDirective.prototype.state;
    /** @type {?} */
    ShCtlStateDirective.prototype.onlyField;
    /**
     * @type {?}
     * @private
     */
    ShCtlStateDirective.prototype.el;
}
