// sh-form-control/esm5/lib/directives/sh-flex.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, Input } from '@angular/core';
var ShFlexDirective = /** @class */ (function () {
    function ShFlexDirective(el) {
        this.el = el;
        /**
         * 最大
         */
        this.colLg = 16.6;
        this.ctlLgFlex = "1 1 " + this.colLg + "%";
        this.ctlLgMaxWidth = this.colLg + "%";
        /**
         * ipad ~ p
         */
        this.colMd = 33.3;
        this.ctlMdFlex = "1 1 " + this.colMd + "%";
        this.ctlMdMaxWidth = this.colMd + "%";
        /**
         * mobile
         */
        this.colSm = 100;
        this.ctlSmFlex = "1 1 " + this.colSm + "%";
        this.ctlSmMaxWidth = this.colSm + "%";
        //メディアクエリ閾値
        this.mediaSm = "(max-width: 768px)";
        this.mediaMd = "(max-width: 1024px)";
        //その他設定しておきたいスタイル
        this.ctlWidth = "100%";
        this.ctlBoxSizing = "border-box";
        this.ctlPadding = "0px 1rem 1rem 0px";
    }
    /**
     * @return {?}
     */
    ShFlexDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        //上から任意の値を設定した場合ここで値を再設定しておかないと
        //意図した挙動にならないので消さない
        this.initParam();
        this.initStyle();
        window.addEventListener('resize', (/**
         * @return {?}
         */
        function () {
            _this.initStyle();
        }));
    };
    /**
     * @private
     * @return {?}
     */
    ShFlexDirective.prototype.initParam = /**
     * @private
     * @return {?}
     */
    function () {
        this.ctlLgFlex = "1 1 " + this.colLg + "%";
        this.ctlLgMaxWidth = this.colLg + "%";
        this.ctlMdFlex = "1 1 " + this.colMd + "%";
        this.ctlMdMaxWidth = this.colMd + "%";
        this.ctlSmFlex = "1 1 " + this.colSm + "%";
        this.ctlSmMaxWidth = this.colSm + "%";
        this.el.nativeElement.style.padding = this.ctlPadding;
        this.el.nativeElement.style.width = this.ctlWidth;
    };
    /**
     * @private
     * @return {?}
     */
    ShFlexDirective.prototype.initStyle = /**
     * @private
     * @return {?}
     */
    function () {
        /** @type {?} */
        var sm = window.matchMedia(this.mediaSm);
        /** @type {?} */
        var md = window.matchMedia(this.mediaMd);
        if (sm.matches) {
            this.smStyleInit();
        }
        else if (md.matches) {
            this.mdStyleInit();
        }
        else {
            this.defaultStyleInit();
        }
    };
    /**
     * @private
     * @return {?}
     */
    ShFlexDirective.prototype.smStyleInit = /**
     * @private
     * @return {?}
     */
    function () {
        this.el.nativeElement.style.padding = this.ctlPadding;
        this.el.nativeElement.style.width = this.ctlWidth;
        this.el.nativeElement.style.flex = this.ctlSmFlex;
        this.el.nativeElement.style.boxSizing = this.ctlBoxSizing;
        this.el.nativeElement.style.maxWidth = this.ctlSmMaxWidth;
    };
    /**
     * @private
     * @return {?}
     */
    ShFlexDirective.prototype.mdStyleInit = /**
     * @private
     * @return {?}
     */
    function () {
        this.el.nativeElement.style.padding = this.ctlPadding;
        this.el.nativeElement.style.width = this.ctlWidth;
        this.el.nativeElement.style.flex = this.ctlMdFlex;
        this.el.nativeElement.style.boxSizing = this.ctlBoxSizing;
        this.el.nativeElement.style.maxWidth = this.ctlMdMaxWidth;
    };
    /**
     * @private
     * @return {?}
     */
    ShFlexDirective.prototype.defaultStyleInit = /**
     * @private
     * @return {?}
     */
    function () {
        this.el.nativeElement.style.padding = this.ctlPadding;
        this.el.nativeElement.style.width = this.ctlWidth;
        this.el.nativeElement.style.flex = this.ctlLgFlex;
        this.el.nativeElement.style.boxSizing = this.ctlBoxSizing;
        this.el.nativeElement.style.maxWidth = this.ctlLgMaxWidth;
    };
    ShFlexDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[ShFlex]'
                },] }
    ];
    /** @nocollapse */
    ShFlexDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ShFlexDirective.propDecorators = {
        colLg: [{ type: Input }],
        colMd: [{ type: Input }],
        colSm: [{ type: Input }]
    };
    return ShFlexDirective;
}());
export { ShFlexDirective };
if (false) {
    /**
     * 最大
     * @type {?}
     */
    ShFlexDirective.prototype.colLg;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlLgFlex;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlLgMaxWidth;
    /**
     * ipad ~ p
     * @type {?}
     */
    ShFlexDirective.prototype.colMd;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlMdFlex;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlMdMaxWidth;
    /**
     * mobile
     * @type {?}
     */
    ShFlexDirective.prototype.colSm;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlSmFlex;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlSmMaxWidth;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.mediaSm;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.mediaMd;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlWidth;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlBoxSizing;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.ctlPadding;
    /**
     * @type {?}
     * @private
     */
    ShFlexDirective.prototype.el;
}
