// sh-form-control/esm5/lib/directives/sh-readonly.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Directive, ElementRef, Input } from '@angular/core';
var ShReadOnlyDirective = /** @class */ (function () {
    function ShReadOnlyDirective(el) {
        this.el = el;
        this.isReadOnly = true;
    }
    /**
     * @return {?}
     */
    ShReadOnlyDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.addAttribute();
    };
    /**
     * @return {?}
     */
    ShReadOnlyDirective.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.addAttribute();
    };
    /**
     * @return {?}
     */
    ShReadOnlyDirective.prototype.addAttribute = /**
     * @return {?}
     */
    function () {
        var e_1, _a, e_2, _b, e_3, _c;
        /** @type {?} */
        var qs = this.el.nativeElement.querySelectorAll('input,textarea,select');
        try {
            for (var qs_1 = tslib_1.__values(qs), qs_1_1 = qs_1.next(); !qs_1_1.done; qs_1_1 = qs_1.next()) {
                var e = qs_1_1.value;
                e.readOnly = this.isReadOnly;
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (qs_1_1 && !qs_1_1.done && (_a = qs_1.return)) _a.call(qs_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
        qs = this.el.nativeElement.querySelectorAll('button,mat-icon');
        if (this.isReadOnly) {
            try {
                for (var qs_2 = tslib_1.__values(qs), qs_2_1 = qs_2.next(); !qs_2_1.done; qs_2_1 = qs_2.next()) {
                    var e = qs_2_1.value;
                    e.style.display = "none";
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (qs_2_1 && !qs_2_1.done && (_b = qs_2.return)) _b.call(qs_2);
                }
                finally { if (e_2) throw e_2.error; }
            }
        }
        else {
            try {
                for (var qs_3 = tslib_1.__values(qs), qs_3_1 = qs_3.next(); !qs_3_1.done; qs_3_1 = qs_3.next()) {
                    var e = qs_3_1.value;
                    e.style.display = "inline";
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (qs_3_1 && !qs_3_1.done && (_c = qs_3.return)) _c.call(qs_3);
                }
                finally { if (e_3) throw e_3.error; }
            }
        }
    };
    ShReadOnlyDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[readonly]'
                },] }
    ];
    /** @nocollapse */
    ShReadOnlyDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ShReadOnlyDirective.propDecorators = {
        isReadOnly: [{ type: Input, args: ['readonly',] }]
    };
    return ShReadOnlyDirective;
}());
export { ShReadOnlyDirective };
if (false) {
    /** @type {?} */
    ShReadOnlyDirective.prototype.isReadOnly;
    /**
     * @type {?}
     * @private
     */
    ShReadOnlyDirective.prototype.el;
}
