// sh-form-control/esm5/lib/components/sh-ctl-check/sh-ctl-check.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { SHFormControl } from '../../utilities/ShFormControl';
var SHCtlCheckComponent = /** @class */ (function (_super) {
    tslib_1.__extends(SHCtlCheckComponent, _super);
    function SHCtlCheckComponent() {
        var _this = _super.call(this) || this;
        //ラベル名
        _this.checkedLabel = "";
        //ラベル位置(チェックボックスの前か)
        _this.beforeAlign = false;
        //チェックボックスのチェック時の値
        _this.checkedValue = "1";
        //チェックボックスの未チェック時の値
        _this.uncheckedValue = "0";
        _this.ctlName = "SHCtlCheck";
        return _this;
    }
    /**
     * @return {?}
     */
    SHCtlCheckComponent.prototype.setLabelPosition = /**
     * @return {?}
     */
    function () {
        return this.beforeAlign ? 'before' : 'after';
    };
    /**
     * @return {?}
     */
    SHCtlCheckComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.onInit();
    };
    /**
     * ngOnChangesイベント
     */
    /**
     * ngOnChangesイベント
     * @param {?} changes
     * @return {?}
     */
    SHCtlCheckComponent.prototype.ngOnChanges = /**
     * ngOnChangesイベント
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes.value) {
            if (changes.value.currentValue) {
                if (changes.value.currentValue === '' || changes.value.currentValue === undefined || changes.value.currentValue === String(/[^\x01-\x7E\xA1-\xDF]+$/g)) {
                    this.value = '';
                    return;
                }
                this.value = changes.value.currentValue == this.checkedValue ? "checked" : "";
            }
        }
    };
    /**
     * onClick
     * @param event click
     */
    /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    SHCtlCheckComponent.prototype.onClick = /**
     * onClick
     * @param {?} event click
     * @return {?}
     */
    function (event) {
        this.onBeforeClick(event);
        this.onAfterClick(event);
    };
    /**
     * onChange
     * @param event change
     */
    /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    SHCtlCheckComponent.prototype.onChange = /**
     * onChange
     * @param {?} event change
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeChange(event);
        this.valueChange.emit(event.checked ? this.checkedValue : this.uncheckedValue);
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterChange(event); }), 0);
    };
    /**
     * onFocus
     * @param event focus
     */
    /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    SHCtlCheckComponent.prototype.onFocus = /**
     * onFocus
     * @param {?} event focus
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeFocus(event);
        event.target.select();
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterFocus(event); }), 0);
    };
    /**
     * onBlur
     * @param event blur
     */
    /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    SHCtlCheckComponent.prototype.onBlur = /**
     * onBlur
     * @param {?} event blur
     * @return {?}
     */
    function (event) {
        var _this = this;
        this.onBeforeBlur(event);
        setTimeout((/**
         * @return {?}
         */
        function () { _this.onAfterBlur(event); }), 0);
        ;
    };
    SHCtlCheckComponent.decorators = [
        { type: Component, args: [{
                    selector: 'sh-ctl-check',
                    template: "<ng-container *ngIf=\"!onlyField; else onlyFieldControl\">\r\n  <div *ngIf=\"!isDisabled\" style=\"height: 52.5px;\">\r\n    <div style=\"height: 52.5px;\">\r\n      <label class=\"sh-label\">{{title}}</label>\r\n      <mat-checkbox class=\"can-moved-with-the-enter-key {{strClass}}\" [(ngModel)]=\"value\" \r\n        [labelPosition]=\"setLabelPosition()\" attr.name=\"{{ctlName}}\" [disabled]=\"disabled\"\r\n        (click)=\"onClick($event)\" (change)=\"onChange($event)\" (focusin)=\"onFocus($event)\" (focusout)=\"onBlur($event)\"\r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\">{{checkedLabel}}\r\n      </mat-checkbox>\r\n    </div>\r\n    <span class=\"custom-alert-message\" style=\"top:0px\">{{message.toString().split(',').join(' ')}}</span>\r\n  </div>\r\n</ng-container>\r\n\r\n<ng-template #onlyFieldControl>\r\n  <div class=\"only-field\">\r\n    <div>\r\n      <mat-checkbox class=\"can-moved-with-the-enter-key {{strClass}}\" [(ngModel)]=\"value\" \r\n        [labelPosition]=\"setLabelPosition()\" attr.name=\"{{ctlName}}\" [disabled]=\"disabled\"\r\n        (click)=\"onClick($event)\" (change)=\"onChange($event)\" (focusin)=\"onFocus($event)\" (focusout)=\"onBlur($event)\"\r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\">{{checkedLabel}}\r\n      </mat-checkbox>\r\n    </div>\r\n  </div>\r\n</ng-template>\r\n\r\n<div *ngIf=\"isDisabled\" class=\"display-ctl-view\">\r\n  <div>\r\n    <div *ngIf=\"title\" class=\"display-ctl-view\">\r\n      <div class=\"display-result\" [ngStyle]=\"{'width.px': labelWidth}\" fxLayout=\"row wrap\" fxFlex>\r\n        <div fxFlex class=\"item-title\"><strong>{{title}}:</strong></div>\r\n      </div>\r\n      <div fxFlex class=\"valueOutput\">\r\n        <mat-checkbox class=\"can-moved-with-the-enter-key {{strClass}}\" [(ngModel)]=\"value\" \r\n        [labelPosition]=\"setLabelPosition()\" \r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\" [disabled]=\"'true'\">{{checkedLabel}}\r\n        </mat-checkbox>\r\n      </div>\r\n    </div>\r\n    <div *ngIf=\"!title\" class=\"display-ctl-view notitle-value\">\r\n      <div fxFlex class=\"valueOutput\">\r\n        <mat-checkbox class=\"can-moved-with-the-enter-key {{strClass}}\" [(ngModel)]=\"value\" \r\n        [labelPosition]=\"setLabelPosition()\" \r\n        [ngStyle]=\"styles\" [ngClass]=\"objClass\" [disabled]=\"'true'\">{{checkedLabel}}\r\n        </mat-checkbox>\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n",
                    styles: ["mat-card-actions,mat-card-content,mat-card-title{text-align:center}mat-card-content{padding-top:20px;padding-bottom:20px}.advanced-color{color:#3f51b5}.beginner-color{color:#f44336}.display-ctl-view{display:-webkit-box;display:flex}.display-result{margin-left:30px}.valueOutput{-webkit-box-flex:1;flex:1}.validation-error::ng-deep .mat-form-field-label{color:red}.validation-error::ng-deep .mat-form-field-underline{background-color:red}.notitle-value{width:100%;margin-left:30px}.sh-label{display:block;color:#0000008a;font-size:.75rem;min-height:18px}", ".sh-flex{width:100%;display:-webkit-inline-box;display:inline-flex}.sh-flex-center{-webkit-box-align:center;align-items:center}.sh-flex-between{-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:center;align-items:center}.display-ctl-view{margin-top:20px}.overlay{position:fixed;display:-webkit-box;display:flex;top:0;left:0;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:center;justify-content:center;-webkit-box-align:center;align-items:center;height:100%;width:100%;color:gray;background-color:rgba(0,0,0,.5);z-index:9999}.loader-text{display:block;color:#f3f3f3;margin-top:1em}.mat-dialog-content{overflow:hidden;max-height:none}.sh-dialog-content{height:85%}.sh-kbn-dialog-table-container{height:92%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-dialog-table-container{display:-webkit-box;display:flex;height:83%;overflow-y:scroll;margin-top:18px;margin-bottom:18px}.sh-text-center{text-align:center}.sh-width-100{width:100%}.sh-label{display:block}.custom-alert-message{top:-20px;position:relative;font-size:.75em;color:rgba(0,0,0,.54)}.result-input,.sh-label{color:rgba(0,0,0,.87)}::ng-deep .mat-form-field-label span{color:rgba(0,0,0,.87)}::ng-deep .result-container .mat-form-field-flex{background-color:#ccdcff}.sh-ctl-state-primary::ng-deep .mat-form-field-label{color:#3f51b5}.sh-ctl-state-primary::ng-deep .mat-form-field-underline{background-color:#3f51b5}.sh-ctl-state-primary-message{color:#3f51b5}.sh-ctl-state-secondary::ng-deep .mat-form-field-label{color:#ff4081}.sh-ctl-state-secondary::ng-deep .mat-form-field-underline{background-color:#ff4081}.sh-ctl-state-secondary-message{color:#ff4081}.sh-ctl-state-success::ng-deep .mat-form-field-label{color:#0dad17}.sh-ctl-state-success::ng-deep .mat-form-field-underline{background-color:#0dad17}.sh-ctl-state-success-message{color:#0dad17}.sh-ctl-state-danger::ng-deep .mat-form-field-label{color:#f44336}.sh-ctl-state-danger::ng-deep .mat-form-field-underline{background-color:#f44336}.sh-ctl-state-danger-message{color:#f44336}.sh-ctl-state-warning::ng-deep .mat-form-field-label{color:#e29f2b}.sh-ctl-state-warning::ng-deep .mat-form-field-underline{background-color:#e29f2b}.sh-ctl-state-warning-message{color:#e29f2b}.sh-container{position:relative;width:100%;height:100%}.only-field{display:-webkit-box;display:flex;-webkit-box-align:center;align-items:center;width:100%;height:100%;min-height:28px}.only-field input,.only-field textarea{font:inherit;background:0 0;color:currentColor;border:none;outline:0;padding:0;margin:0;width:100%;max-width:100%;vertical-align:bottom;text-align:inherit}.only-field-danger{background-color:#f44336}.only-field-warning{background-color:#e29f2b}.only-field-success{background-color:#0dad17}::ng-deep .only-field .material-icons{font-size:13px}::ng-deep .only-field .mat-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon-button{position:relative;width:24px;height:24px}::ng-deep .only-field .mat-icon{position:absolute;top:0;left:0}::ng-deep .only-field .mat-datepicker-toggle-default-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;margin-right:auto}::ng-deep .mat-datepicker-toggle span svg{width:16px;height:16px}.only-label{background-color:#ccdcff;position:relative}.only-label mat-icon{position:absolute;top:25%;left:0;right:0;bottom:0;margin-left:auto;font-size:16px}.bind-flg__input{opacity:0;height:0;width:0;position:absolute;z-index:-100}"]
                }] }
    ];
    /** @nocollapse */
    SHCtlCheckComponent.ctorParameters = function () { return []; };
    SHCtlCheckComponent.propDecorators = {
        checkedLabel: [{ type: Input }],
        beforeAlign: [{ type: Input }],
        checkedValue: [{ type: Input }],
        uncheckedValue: [{ type: Input }]
    };
    return SHCtlCheckComponent;
}(SHFormControl));
export { SHCtlCheckComponent };
if (false) {
    /** @type {?} */
    SHCtlCheckComponent.prototype.checkedLabel;
    /** @type {?} */
    SHCtlCheckComponent.prototype.beforeAlign;
    /** @type {?} */
    SHCtlCheckComponent.prototype.checkedValue;
    /** @type {?} */
    SHCtlCheckComponent.prototype.uncheckedValue;
}
