// sh-http-client/esm2015/lib/services/alert-handler.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { Alert } from '../utilities/Alert';
import * as i0 from "@angular/core";
export class AlertHandlerService {
    /**
     * コンストラクタ
     */
    constructor() {
        this.Alert = new Alert();
        this.AlertHandlerSubject = new Subject();
        this.AlertHandlerState = this.AlertHandlerSubject.asObservable();
    }
    /**
     * アラートを追加する
     * @param {?} alertInfo
     * @return {?}
     */
    setAlert(alertInfo) {
        this.Alert = alertInfo;
        this.AlertHandlerSubject.next(this.Alert);
    }
    /**
     * アラートをクリアする
     * @return {?}
     */
    alertClear() {
        this.Alert.message = "";
        this.Alert.type = "";
        this.AlertHandlerSubject.next(this.Alert);
    }
}
AlertHandlerService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
AlertHandlerService.ctorParameters = () => [];
/** @nocollapse */ AlertHandlerService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function AlertHandlerService_Factory() { return new AlertHandlerService(); }, token: AlertHandlerService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    AlertHandlerService.prototype.Alert;
    /**
     * @type {?}
     * @private
     */
    AlertHandlerService.prototype.AlertHandlerSubject;
    /** @type {?} */
    AlertHandlerService.prototype.AlertHandlerState;
}
