// sh-http-client/esm2015/lib/services/logger.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShHttpClientService } from './sh-http-client.service';
import { Router, NavigationEnd } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "./sh-http-client.service";
import * as i2 from "@angular/router";
export class LoggerService {
    /**
     * @param {?} http
     * @param {?} router
     */
    constructor(http, router) {
        this.http = http;
        this.router = router;
        this.isBackButton = false;
        this.config = { url: '', useHash: true };
    }
    /**
     * logApiへリクエスト
     *
     * @private
     * @param {?} param
     * @return {?}
     */
    push(param) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return this.http.post(this.config.url, param);
        });
    }
    /**
     * 設定を更新
     * @template THIS
     * @this {THIS}
     * @param {?} config
     * @return {THIS}
     */
    configure(config) {
        const { url, useHash } = config;
        if (url) {
            (/** @type {?} */ (this)).config.url = url;
        }
        if (useHash !== undefined && useHash !== null) {
            (/** @type {?} */ (this)).config.useHash = useHash;
        }
        return (/** @type {?} */ (this));
    }
    /**
     * 画面遷移が発生するたびにlogging
     * @return {?}
     */
    trackingPath() {
        this.router.events.subscribe((/**
         * @param {?} event
         * @return {?}
         */
        (event) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            if (event instanceof NavigationEnd) {
                if (this.isBackButton) {
                    this.isBackButton = false;
                    return;
                }
                /** @type {?} */
                const path = this.config.useHash
                    ? `#${event.urlAfterRedirects}`
                    : event.urlAfterRedirects.replace('/', '');
                /** @type {?} */
                const result = yield this.push({ path });
            }
        })));
    }
    /**
     * actionlog pathは必須のため必ず送られる
     * @param {?} action
     * @return {?}
     */
    action(action) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const path = this.config.useHash
                ? `#${this.router.url}`
                : this.router.url.replace('/', '');
            yield this.push({ path, action });
        });
    }
    /**
     *
     * @return {?}
     */
    stopLoggingPath() {
        this.isBackButton = true;
    }
}
LoggerService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
LoggerService.ctorParameters = () => [
    { type: ShHttpClientService },
    { type: Router }
];
/** @nocollapse */ LoggerService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function LoggerService_Factory() { return new LoggerService(i0.ɵɵinject(i1.ShHttpClientService), i0.ɵɵinject(i2.Router)); }, token: LoggerService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.isBackButton;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.config;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.http;
    /**
     * @type {?}
     * @private
     */
    LoggerService.prototype.router;
}
