// sh-http-client/esm2015/lib/services/navigation.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class NavigationService {
    /**
     * コンストラクタ
     */
    constructor() {
        //ユーザーのブラウザバック実行状態
        this.isBackClicked = false;
    }
    /**
     * ユーザーのブラウザバック実行を保存する
     * @return {?} true/false
     */
    getBackClicked() {
        return this.isBackClicked;
    }
    /**
     * ユーザーのブラウザバック実行を保存する
     * @param {?} isBackClicked
     * @return {?}
     */
    setBackClicked(isBackClicked) {
        this.isBackClicked = isBackClicked;
    }
}
NavigationService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
NavigationService.ctorParameters = () => [];
/** @nocollapse */ NavigationService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NavigationService_Factory() { return new NavigationService(); }, token: NavigationService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    NavigationService.prototype.isBackClicked;
}
