// sh-http-client/esm2015/lib/services/sh-redirect.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, InjectionToken, Inject } from '@angular/core';
import { Router } from '@angular/router';
import { ShHttpClientConst } from '../constShHttpClient';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/** @type {?} */
export const SH_BROWSER_STORAGE = new InjectionToken(`Browser Storage`, {
    providedIn: `root`,
    factory: (/**
     * @return {?}
     */
    () => localStorage)
});
/** @type {?} */
export const USE_HASH = new InjectionToken('UseHash', {
    providedIn: `root`,
    factory: (/**
     * @return {?}
     */
    () => ({ useHash: true }))
});
export class ShRedirectService {
    /**
     * @param {?} router
     * @param {?} storage
     * @param {?} isHash
     */
    constructor(router, storage, isHash) {
        this.router = router;
        this.storage = storage;
        this.isHash = isHash;
    }
    /**
     * @return {?}
     */
    get redirectPath() {
        return this.storage.getItem(ShHttpClientConst.REDIRECT_PATH);
    }
    /**
     * @return {?}
     */
    get queryParams() {
        /** @type {?} */
        const query = new URLSearchParams(this.storage.getItem(ShHttpClientConst.SH_QUERY_PARAMS)) || null;
        return query;
    }
    /**
     * @return {?}
     */
    setRedirectPath() {
        /** @type {?} */
        let path = '';
        /** @type {?} */
        let query = '';
        if (this.isHash.useHash) {
            /** @type {?} */
            const pathparams = location.href.split('#')[1].split('?');
            path = pathparams[0];
            query = pathparams[1] || '';
        }
        else {
            path = location.pathname;
            path = location.search;
        }
        this.storage.setItem(ShHttpClientConst.REDIRECT_PATH, path);
        this.storage.setItem(ShHttpClientConst.SH_QUERY_PARAMS, query);
    }
    /**
     * @return {?}
     */
    redirect() {
        this.router.navigate([this.storage.getItem(ShHttpClientConst.REDIRECT_PATH)]);
        this.storage.setItem(ShHttpClientConst.REDIRECT_PATH, '');
    }
}
ShRedirectService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ShRedirectService.ctorParameters = () => [
    { type: Router },
    { type: Storage, decorators: [{ type: Inject, args: [SH_BROWSER_STORAGE,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [USE_HASH,] }] }
];
/** @nocollapse */ ShRedirectService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ShRedirectService_Factory() { return new ShRedirectService(i0.ɵɵinject(i1.Router), i0.ɵɵinject(SH_BROWSER_STORAGE), i0.ɵɵinject(USE_HASH)); }, token: ShRedirectService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.router;
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.storage;
    /**
     * @type {?}
     * @private
     */
    ShRedirectService.prototype.isHash;
}
