// sh-http-client/esm2015/lib/services/validation-error.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
class ValidationError {
}
if (false) {
    /** @type {?} */
    ValidationError.prototype.show;
    /** @type {?} */
    ValidationError.prototype.meesages;
}
export class ValidationErrorService {
    constructor() {
        this.validationError = new ValidationError();
        this.validationErrorSubject = new Subject();
        this.validationErrorState = this.validationErrorSubject.asObservable();
    }
    /**
     * @param {?} validationErrors
     * @return {?}
     */
    setValidationError(validationErrors) {
        this.validationError.show = true;
        this.validationError.meesages = validationErrors;
        this.validationErrorSubject.next(this.validationError);
    }
    /**
     * @return {?}
     */
    clearValidationError() {
        this.validationError.show = false;
        this.validationError.meesages = null;
        this.validationErrorSubject.next(this.validationError);
    }
}
ValidationErrorService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ValidationErrorService.ctorParameters = () => [];
/** @nocollapse */ ValidationErrorService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ValidationErrorService_Factory() { return new ValidationErrorService(); }, token: ValidationErrorService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    ValidationErrorService.prototype.validationError;
    /**
     * @type {?}
     * @private
     */
    ValidationErrorService.prototype.validationErrorSubject;
    /** @type {?} */
    ValidationErrorService.prototype.validationErrorState;
}
