// sh-http-client/esm2015/lib/utilities/ShMessage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class SHMessageUtl {
    /**
     * テンプレート文字列　形式：{0}{1}{2}
     * @param {?} text
     * @param {...?} values
     * @return {?}
     */
    stringFormat(text, ...values) {
        /** @type {?} */
        let result = text;
        values.forEach((/**
         * @param {?} value
         * @param {?} i
         * @return {?}
         */
        function (value, i) {
            result = result.split(`{${i}}`).join(value);
        }));
        return result;
    }
}
export class SHMessageConfirm extends SHMessageUtl {
    /**
     * 実行確認メッセージ
     * @param {?} text
     * @return {?}
     */
    execution(text) {
        return this.stringFormat("{0}を行います。\nよろしいですか？", text);
    }
    /**
     * 終了確認メッセージ
     * @param {?} text
     * @return {?}
     */
    finalization(text) {
        return this.stringFormat("{0}を終了します。\nよろしいですか？", text);
    }
    /**
     * 登録確認メッセージ
     * @return {?}
     */
    creatingRecord() {
        return this.execution("登録");
    }
    /**
     * 修正確認メッセージ
     * @return {?}
     */
    editingRecord() {
        return this.execution("修正");
    }
    /**
     * 削除確認メッセージ
     * @return {?}
     */
    deletingRecord() {
        return this.execution("削除");
    }
    /**
     * 更新確認メッセージ
     * @return {?}
     */
    updatingRecord() {
        return this.execution("更新");
    }
    /**
     * 印刷確認メッセージ
     * @return {?}
     */
    printingOut() {
        return this.execution("印刷");
    }
    /**
     * 編集内容が失われる警告メッセージ
     * @return {?}
     */
    editingContentWillBeLost() {
        return "編集内容が失われます。\nよろしいですか？";
    }
}
export class SHMessageFatal extends SHMessageUtl {
    /**
     * システムエラーメッセージ
     * @return {?}
     */
    systemErrorHasOccurred() {
        return "システムエラーが発生しました。システム管理者に連絡してください。";
    }
}
export class SHMessageInfo extends SHMessageUtl {
    /**
     * 前のページに戻る場合の確認メッセージ
     * @param {?} btnName
     * @return {?}
     */
    dontUseHistoryBack(btnName) {
        return this.stringFormat("前のページに戻る場合、{0}ボタンから戻ってください。", btnName);
    }
}
export class SHMessage {
    constructor() {
        /**
         * Confirm
         */
        this.confirm = new SHMessageConfirm();
        /**
         * Fatal
         */
        this.fatal = new SHMessageFatal();
        /**
         * Info
         */
        this.info = new SHMessageInfo();
    }
}
if (false) {
    /**
     * Confirm
     * @type {?}
     */
    SHMessage.prototype.confirm;
    /**
     * Fatal
     * @type {?}
     */
    SHMessage.prototype.fatal;
    /**
     * Info
     * @type {?}
     */
    SHMessage.prototype.info;
}
