// sh-form-control/esm5/lib/services/bind-ctl.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import { ShCtlConst } from "../constShCtl";
import { NoChangeFlg } from '../types/NoChangeFlg';
import * as i0 from "@angular/core";
var BindCtlService = /** @class */ (function () {
    function BindCtlService() {
        this.NO_CHANGE_FLG_SUFFIX = ShCtlConst.NO_CHANGE.FLG_SUFFIX;
    }
    /**
     * @param {?} param
     * @return {?}
     */
    BindCtlService.prototype.setNoChangeFlg = /**
     * @param {?} param
     * @return {?}
     */
    function (param) {
        var _this = this;
        if (this.isParamEmpty(param))
            return;
        /** @type {?} */
        var info = this.generateBindInputInfo(param);
        info.forEach((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            /** @type {?} */
            var targetInput = (/** @type {?} */ (document.querySelector("input[name=" + data.ctlName + "_" + _this.NO_CHANGE_FLG_SUFFIX + "],textarea[name=" + data.ctlName + "_" + _this.NO_CHANGE_FLG_SUFFIX + "]")));
            if (!targetInput)
                return;
            targetInput.value = NoChangeFlg.ON;
            targetInput.dispatchEvent(new Event("change"));
        }));
    };
    /**
     * @param {?} param
     * @return {?}
     */
    BindCtlService.prototype.setBindValue = /**
     * @param {?} param
     * @return {?}
     */
    function (param) {
        if (this.isParamEmpty(param))
            return;
        /** @type {?} */
        var info = this.generateBindInputInfo(param);
        info.forEach((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            /** @type {?} */
            var targetInput = (/** @type {?} */ (document.querySelector("input[name=" + data.ctlName + "],textarea[name=" + data.ctlName + "]")));
            if (!targetInput)
                return;
            targetInput.value = data.value;
            targetInput.dispatchEvent(new Event("change"));
        }));
    };
    /**
     * @private
     * @param {?} param
     * @return {?}
     */
    BindCtlService.prototype.generateBindInputInfo = /**
     * @private
     * @param {?} param
     * @return {?}
     */
    function (param) {
        return param.bindColumnArray.map((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            /** @type {?} */
            var column = Number(Object.values(data)[0]);
            return {
                ctlName: Object.keys(data)[0],
                value: isNaN(Number(column)) ? "" : (/** @type {?} */ (Object.values(param.record)[column - 1])),
            };
        }));
    };
    /**
     * @private
     * @param {?} param
     * @return {?}
     */
    BindCtlService.prototype.isParamEmpty = /**
     * @private
     * @param {?} param
     * @return {?}
     */
    function (param) {
        if (!param.bindColumnArray || param.bindColumnArray.length == 0)
            return true;
        if (!param.record)
            return true;
        return false;
    };
    BindCtlService.decorators = [
        { type: Injectable, args: [{
                    providedIn: "root",
                },] }
    ];
    /** @nocollapse */
    BindCtlService.ctorParameters = function () { return []; };
    /** @nocollapse */ BindCtlService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function BindCtlService_Factory() { return new BindCtlService(); }, token: BindCtlService, providedIn: "root" });
    return BindCtlService;
}());
export { BindCtlService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    BindCtlService.prototype.NO_CHANGE_FLG_SUFFIX;
}
