// sh-form-control/esm5/lib/services/sh-ctl-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShHttpClientService } from 'sh-http-client';
import * as i0 from "@angular/core";
import * as i1 from "sh-http-client";
/**
 * ShCtl　Api通信クラス
 */
var ShCtlApiService = /** @class */ (function () {
    function ShCtlApiService(apiComService) {
        this.apiComService = apiComService;
    }
    /**
    * API返却結果に応じて処理分岐を行い、ApplicationDataのみを返却する
    * @param apiResponse API返却結果
    * @return API返却結果のApplicationData
    */
    /**
     * API返却結果に応じて処理分岐を行い、ApplicationDataのみを返却する
     * @private
     * @template T
     * @param {?} apiResponseBody
     * @return {?} API返却結果のApplicationData
     */
    ShCtlApiService.prototype.getApplicationData = /**
     * API返却結果に応じて処理分岐を行い、ApplicationDataのみを返却する
     * @private
     * @template T
     * @param {?} apiResponseBody
     * @return {?} API返却結果のApplicationData
     */
    function (apiResponseBody) {
        //バリデーション処理
        //アラート処理
        //システムに返す際は処理に必要なデータだけにする
        return apiResponseBody.applicationData;
    };
    /**
    * GET
    * @param url ApiUrl
    * @return API返却結果
    */
    /**
     * GET
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    ShCtlApiService.prototype.get = /**
     * GET
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    function (url) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var ret;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        ret = null;
                        return [4 /*yield*/, this.apiComService.get(url, false)
                                .then((/**
                             * @param {?} response
                             * @return {?}
                             */
                            function (response) {
                                ret = _this.getApplicationData(response);
                            }))];
                    case 1:
                        _a.sent();
                        return [2 /*return*/, ret];
                }
            });
        });
    };
    /**
    * POST
    * @param url ApiUrl
    * @param data データ
    * @return API返却結果
    */
    /**
     * POST
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    ShCtlApiService.prototype.post = /**
     * POST
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    function (url, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var ret;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        ret = null;
                        return [4 /*yield*/, this.apiComService.post(url, data)
                                .then((/**
                             * @param {?} response
                             * @return {?}
                             */
                            function (response) {
                                ret = _this.getApplicationData(response);
                            }))];
                    case 1:
                        _a.sent();
                        return [2 /*return*/, ret];
                }
            });
        });
    };
    /**
    * PUT
    * @param url ApiUrl
    * @param data データ
    * @return API返却結果
    */
    /**
     * PUT
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    ShCtlApiService.prototype.put = /**
     * PUT
     * @template T
     * @param {?} url ApiUrl
     * @param {?} data データ
     * @return {?} API返却結果
     */
    function (url, data) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var ret;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        ret = null;
                        return [4 /*yield*/, this.apiComService.put(url, data)
                                .then((/**
                             * @param {?} response
                             * @return {?}
                             */
                            function (response) {
                                ret = _this.getApplicationData(response);
                            }))];
                    case 1:
                        _a.sent();
                        return [2 /*return*/, ret];
                }
            });
        });
    };
    /**
    * DELETE
    * @param url ApiUrl
    * @return API返却結果
    */
    /**
     * DELETE
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    ShCtlApiService.prototype.delete = /**
     * DELETE
     * @template T
     * @param {?} url ApiUrl
     * @return {?} API返却結果
     */
    function (url) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var ret;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        ret = null;
                        return [4 /*yield*/, this.apiComService.delete(url)
                                .then((/**
                             * @param {?} response
                             * @return {?}
                             */
                            function (response) {
                                ret = _this.getApplicationData(response);
                            }))];
                    case 1:
                        _a.sent();
                        return [2 /*return*/, ret];
                }
            });
        });
    };
    ShCtlApiService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ShCtlApiService.ctorParameters = function () { return [
        { type: ShHttpClientService }
    ]; };
    /** @nocollapse */ ShCtlApiService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ShCtlApiService_Factory() { return new ShCtlApiService(i0.ɵɵinject(i1.ShHttpClientService)); }, token: ShCtlApiService, providedIn: "root" });
    return ShCtlApiService;
}());
export { ShCtlApiService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ShCtlApiService.prototype.apiComService;
}
