// sh-form-control/esm5/lib/services/smpl-kbn-code.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { ShCtlApiService } from './sh-ctl-api.service';
import * as i0 from "@angular/core";
import * as i1 from "./sh-ctl-api.service";
/**
 * 区分マスタの情報を取得
 */
var SmplKbnCodeService = /** @class */ (function () {
    function SmplKbnCodeService(apiComService) {
        this.apiComService = apiComService;
        //区分マスタ
        this.masterCache = [];
    }
    /**
     * 区分マスタ取得
     */
    /**
     * 区分マスタ取得
     * @param {?} url
     * @return {?}
     */
    SmplKbnCodeService.prototype.fetchAll = /**
     * 区分マスタ取得
     * @param {?} url
     * @return {?}
     */
    function (url) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                return [2 /*return*/, this.apiComService.get(url)];
            });
        });
    };
    /**
     * 指定したマスタのキャッシュしているデータを取得
     * @param key マスタの名前
     */
    /**
     * 指定したマスタのキャッシュしているデータを取得
     * @param {?} key マスタの名前
     * @return {?}
     */
    SmplKbnCodeService.prototype.getMasterData = /**
     * 指定したマスタのキャッシュしているデータを取得
     * @param {?} key マスタの名前
     * @return {?}
     */
    function (key) {
        /** @type {?} */
        var cache = this.masterCache.filter((/**
         * @param {?} item
         * @return {?}
         */
        function (item) {
            return item.masterName == key;
        }));
        return cache.length > 0 ? cache[0].masterCache : [];
    };
    /**
     * マスタデータをキャッシュする
     * @param key マスタの名前
     * @param masterData マスタデータ
     */
    /**
     * マスタデータをキャッシュする
     * @param {?} key マスタの名前
     * @param {?} masterData マスタデータ
     * @return {?}
     */
    SmplKbnCodeService.prototype.setMasterData = /**
     * マスタデータをキャッシュする
     * @param {?} key マスタの名前
     * @param {?} masterData マスタデータ
     * @return {?}
     */
    function (key, masterData) {
        var e_1, _a;
        if (this.masterCache.length > 0) {
            try {
                for (var _b = tslib_1.__values(this.masterCache), _c = _b.next(); !_c.done; _c = _b.next()) {
                    var item = _c.value;
                    if (item.masterName == key) {
                        item.masterCache = masterData;
                        break;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
        }
        this.masterCache = this.masterCache.concat([{ masterName: key, masterCache: masterData }]);
    };
    SmplKbnCodeService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    SmplKbnCodeService.ctorParameters = function () { return [
        { type: ShCtlApiService }
    ]; };
    /** @nocollapse */ SmplKbnCodeService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function SmplKbnCodeService_Factory() { return new SmplKbnCodeService(i0.ɵɵinject(i1.ShCtlApiService)); }, token: SmplKbnCodeService, providedIn: "root" });
    return SmplKbnCodeService;
}());
export { SmplKbnCodeService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    SmplKbnCodeService.prototype.masterCache;
    /**
     * @type {?}
     * @private
     */
    SmplKbnCodeService.prototype.apiComService;
}
